---
title: "FormatMultiQC"
author: "Shelly Trigg"
date: "7/16/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

load libraries
```{r}
library(data.table)
library(gtools)
library(ggplot2)
library(knitr)
library(kableExtra)
```

read in data 
```{r}
#cutadapt stats
multiqc_a <- data.frame(fread("https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200716/multiqc_data/multiqc_cutadapt.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE))

#bismark
multiqc_b <- data.frame(fread("https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200716/multiqc_data/multiqc_bismark_alignment.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE))

#bismark meth-extract
multiqc_c <- data.frame(fread("https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200716/multiqc_data/multiqc_bismark_methextract.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE))

#bismark dedup
multiqc_d <- data.frame(fread("https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200716/multiqc_data/multiqc_bismark_dedup.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE))

#general stats
multiqc_g <- data.frame(fread("https://gannet.fish.washington.edu/metacarcinus/Pgenerosa/analyses/20200716/multiqc_data/multiqc_general_stats.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE))

#lambda alignment stats
#multiqc_l <- fread("https://gannet.fish.washington.edu/metacarcinus/FROGER_meth_compare/20200424/multiqc_data/multiqc_bismark_alignment.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)

meta_data <- read.csv("RAnalysis/Data/Sample.Info.csv", stringsAsFactors = FALSE)




```


format meta data
```{r}
```


format multiqc sample names
```{r}
multiqc_a$Sample <- gsub("_S.*_L.*_R", "_R", multiqc_a$Sample)
multiqc_a$Sample <- gsub("_001", "", multiqc_a$Sample)
multiqc_b$Sample <- gsub("_.*", "", multiqc_b$Sample)
multiqc_c$Sample <- gsub("_.*", "", multiqc_c$Sample)
multiqc_d$Sample <- gsub("_.*", "", multiqc_d$Sample)
multiqc_g$Sample <- gsub("_S.*_L.*_R", "_R", multiqc_g$Sample)
multiqc_g$Sample <- gsub("_001", "", multiqc_g$Sample)
multiqc_g$Sample <- gsub("_val.*", "", multiqc_g$Sample)


```


format column names so merging won't produce duplicate colnames
```{r}
colnames(multiqc_b)[2] <- "uniq_aligned_reads"
colnames(multiqc_d)[2] <- "uniq_aligned_reads_no_discard"

```

#create table with trimming stats
```{r}
#subset the general stats table for the fastqc data
multiqc_f <- multiqc_g[,c(grep("Sample|FastQC",colnames(multiqc_g)))]
#remove rows with all NAs
multiqc_f <- multiqc_f[complete.cases(multiqc_f),]


#merge cutadapt and fastqc stats by Sample
trim_multiqc <- merge(multiqc_a,multiqc_f, by = "Sample")

#create read pair column
trim_multiqc$read_pair <- gsub(".*_R","R",trim_multiqc$Sample)

#remove read pair info from sample and rename sample column to Sample.ID
trim_multiqc$Sample.ID <- gsub("_R.*","",trim_multiqc$Sample)
#convert hyphens to underscores
trim_multiqc$Sample.ID <- gsub("-","_",trim_multiqc$Sample.ID)

#remove original sample column
trim_multiqc$Sample <- NULL

#merge with meta data
trim_multiqc <- merge(meta_data[,c("Sample.ID", "Tank","Initial.Treatment","Secondary.Treatment", "TimePoint")], trim_multiqc, by = "Sample.ID")

#move read pair column 
trim_multiqc <- trim_multiqc[,c(1:5,17,6:16)]

#remove extra text from colnames
colnames(trim_multiqc) <- gsub("FastQC_mqc\\.generalstats\\.","", colnames(trim_multiqc))

#order table by Sample.ID 
trim_multiqc <- trim_multiqc[mixedorder(trim_multiqc$Sample.ID),]

write.csv(trim_multiqc, "RAnalysis/Output/Raw_trimmed_data_descriptive_stats.csv", row.names = FALSE, quote = FALSE)

```

create a bismark table
```{r}
#merge dedup and meth extract data
multi_bismark <- merge(multiqc_d, multiqc_c, by = "Sample", all = TRUE)

#merge with bismark alignment report
#exclude meth calls from bismark alignments because better to include these from meth extract report
colnames(multiqc_b[,c(5:7,10:18,20:22)]) #print columns to exclude
multi_bismark <- merge(multiqc_b[,-c(5:7,10:18,20:22)], multi_bismark, by = "Sample", all = TRUE) #merge

#change column 1 to Sample.ID
colnames(multi_bismark)[1] <- "Sample.ID"

#convert hyphen to underscore
multi_bismark$Sample.ID <- gsub("-","_",multi_bismark$Sample.ID)

#merge with meta data
multi_bismark <- merge(meta_data[,c("Sample.ID", "Tank","Initial.Treatment","Secondary.Treatment", "TimePoint")],multi_bismark, by ="Sample.ID")

#order data by sample 
multi_bismark <- multi_bismark[mixedorder(as.character(multi_bismark$Sample.ID)),]

#write out alignments table
write.csv(multi_bismark,"RAnalysis/Output/Alignments_descriptive_stats.csv", row.names = FALSE, quote = FALSE)
```

generate simplified table for manuscript main text
```{r}
multi_bismark_simp <- multi_bismark[,c("Sample","Species", "Sequencing.Method","total_reads","no_alignments", "ambig_reads", "uniq_aligned_reads_no_discard", "dedup_reads")]

#add raw reads column
multi_bismark_simp <- merge(data.frame(unique(trim_multiqc[,c("Sample","Species","Sequencing.Method", "r_processed")])), multi_bismark_simp, by = c("Sample","Species","Sequencing.Method"))

#order by sample no
multi_bismark_simp <- multi_bismark_simp[mixedorder(as.character(multi_bismark_simp$Sample)),]


#fill in RRBS cells with uniq_aligned reads
for(i in 1:nrow(multi_bismark_simp)){
  if(is.na(multi_bismark_simp$uniq_aligned_reads_no_discard[i])){
    multi_bismark_simp$uniq_aligned_reads_no_discard[i] <- multi_bismark$uniq_aligned_reads[i]
  }
  if(is.na(multi_bismark_simp$dedup_reads[i])){
    multi_bismark_simp$dedup_reads[i] <- multi_bismark$uniq_aligned_reads[i]
  }
}

#convert to sci not.
#for(i in 4:ncol(multi_bismark_simp)){
#  multi_bismark_simp[,i] <- formatC(multi_bismark_simp[,i], format = "e", digits = 2)
#}

#put table in terms of M reads
for(i in 4:ncol(multi_bismark_simp)){
  multi_bismark_simp[,i] <- round((multi_bismark_simp[,i]/1000000),2)
}

#remove sample column
multi_bismark_simp$Sample <- NULL

#change MBD ... to just MBD in method column
multi_bismark_simp$Sequencing.Method <- gsub("MBD.*", "MBDBS", multi_bismark_simp$Sequencing.Method)

#rename columns
colnames(multi_bismark_simp) <- c("species", "method", "raw", "trimmed", "unaligned", "ambiguously aligned", "uniquely aligned", "uniquely aligned excluding duplicates")

#pretty format
kable(multi_bismark_simp, row.names = FALSE,align='ccccccc') %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))  %>% row_spec(c(1:3,10:12), color = 'black', background = "#BAE4B3") %>% row_spec(c(4:6,13:15), color = 'black', background = "#CBC9E2") %>% row_spec(c(7:9,16:18), color = 'black', background = '#FDBE85') %>% column_spec (1:8,border_left = T, border_right = T) %>% column_spec(6:8, width = "4cm") %>% as_image(file = "../../Output/Summary_descriptive_statistics_Mcap_Pact.jpg")




```

## plot different stats

percent reads with unique alignment 
```{r}
jpeg("../../Output/Pact_Mcap_boxplot_methodXuniquely_mapped_reads.jpg", width = 8, height= 3, units = "in", res = 300 )
ggplot(multi_bismark) + geom_boxplot(aes(x =Sequencing.Method,y = percent_aligned,group = Sequencing.Method, color = Sequencing.Method,fill = Sequencing.Method), alpha = 0.3) + facet_wrap(~Species) + theme_bw() + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x=element_blank()) + ylab("uniquely mapped reads (%)")
dev.off()
```

percent CpG methylation 
```{r}
jpeg("../../Output/Pact_Mcap_boxplot_methodXmethCpG.jpg", width = 8, height= 3, units = "in", res = 300 )
ggplot(multi_bismark) + geom_boxplot(aes(x =Sequencing.Method,y = percent_cpg_meth,group = Sequencing.Method, color = Sequencing.Method, fill = Sequencing.Method), alpha = 0.5) + facet_wrap(~Species) + theme_bw() + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x=element_blank()) + ylab("CpG methylation (%)")
dev.off()
```

## Other potential plots
percent reads with any alignment (unique or ambig)
```{r}
ggplot(multi_bismark) + geom_boxplot(aes(x =Sequencing.Method,y = ((uniq_aligned_reads + ambig_reads)/total_reads),group = Sequencing.Method, fill = Sequencing.Method)) + facet_wrap(~Species) + theme_bw() + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x=element_blank()) + ylab("mapped reads (%)")

```

percent CHG methylation 
```{r}
ggplot(multi_bismark) + geom_boxplot(aes(x =Sequencing.Method,y = percent_chg_meth,group = Sequencing.Method, color = Sequencing.Method, fill = Sequencing.Method), alpha = 0.5) + facet_wrap(~Species) + theme_bw() + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x=element_blank()) + ylab("CHG methylation (%)")
```

percent CHH methylation 
```{r}
ggplot(multi_bismark) + geom_boxplot(aes(x =Sequencing.Method,y = percent_chh_meth,group = Sequencing.Method, color = Sequencing.Method, fill = Sequencing.Method), alpha = 0.5) + facet_wrap(~Species) + theme_bw() + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x=element_blank()) + ylab("CHH methylation (%)")
```