---
title: "Geoduck DNA Methylation"
author: "HPutnam"
date: "05/14/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Load Libraries

```{r message = FALSE, warning = FALSE}
library(plotrix) 
library(ggplot2)
library(gridExtra)
library(seacarb) 
library(pheatmap)
library(dplyr)
library(tidyverse)
library(tidyr)
library(topGO)
library(goseq)
library(genefilter)
library(gplots)
library(cowplot)
library(lsmeans)
library(ontologyIndex)
library(ontologySimilarity)
library(data.table)
library(RColorBrewer)
library(colorRamps)
```

```{r Flow Rates}
# FLOW RATES
#Initial Exposure: Quantifying flow and testing for differences among treatments
Flow1 <- read.csv("RAnalysis/Data/Flow1_Juvenile_Geoduck.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
#flow values measured in ml/10 sec
Flow1$Rate <- (((Flow1$Rate1 + Flow1$Rate2)/2)*6)/1000*60 #calculate flow rate in liters per hour
Flow1.Rate <- aggregate(Rate ~ Treatment, data=Flow1, FUN=mean) #mean flow by treatment
Flow1.Rate$se <- aggregate(Rate ~ Treatment, data=Flow1, FUN=std.error) #sem flow by treatment
mean(Flow1$Rate) #mean flow across all tanks
std.error(Flow1$Rate) #sem flow across all tanks


FL1 <- aov(Rate ~ Treatment, data=Flow1) #test for differences in flow rate between treatments
FL1.res <- anova(FL1) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(FL1$residuals) #plot histogram of residuals
boxplot(FL1$residuals) #plot boxplot of residuals
plot(FL1) #display residuals versus fitter, normal QQ plot, leverage plot


#Secondary Exposure: Quantifying flow and testing for differences among treatments
Flow2 <- read.csv("RAnalysis/Data/Flow2_Juvenile_Geoduck.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
#flow values measured in ml/15 sec
Flow2$Rate <- ((Flow2$Rate1)*4)/1000*60 #calculate flow rate in liters per hour
Flow2.Rate <- aggregate(Rate ~ Treatment, data=Flow2, FUN=mean) #mean flow by treatment
Flow2.Rate$se <- aggregate(Rate ~ Treatment, data=Flow2, FUN=std.error) #sem flow by treatment
mean(Flow2$Rate) #mean flow across all tanks
std.error(Flow2$Rate) #sem flow across all tanks

FL2 <- aov(Rate ~ Treatment, data=Flow2) #test for differences in flow rate between treatments
FL2.res <-anova(FL2) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(FL2$residuals) #plot histogram of residuals
boxplot(FL2$residuals) #plot boxplot of residuals
plot(FL2) #display residuals versus fitter, normal QQ plot, leverage plot
```

```{r Feeding Rates}
# FEEDING RATES
#Initial Exposure: Quantifying feeding and testing for differences among treatments
cells.1 <- read.csv("RAnalysis/Data/Cell_Counts_Juvenile_Geoduck_Exp1.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
cells.1$Avg.Cells <- rowMeans(cells.1[,c("Count1",  "Count2")], na.rm = TRUE) #calculate average of counts
cells.1$cell.num <- cells.1$Avg.Cells/cells.1$Volume.Counted #calculate density

avg.cells.tank.1 <- do.call(data.frame,aggregate(cell.num ~ Tank, data = cells.1, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Tank
avg.cells.trt.1 <- do.call(data.frame,aggregate(cell.num ~ Treatment, data = cells.1, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Treatment
avg.cells.1 <- do.call(data.frame,aggregate(cell.num ~ Volume.Counted, data = cells.1, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Treatment
colnames(avg.cells.tank.1) <- c("Tank", "mean", "se") #rename columns 
colnames(avg.cells.trt.1) <- c("Treatment", "mean", "se") #rename columns 

cells1.tank <- aov(cell.num ~Tank, data=cells.1) #test the hypothesis feeding does not differ between tanks
cells1.tank.res <-anova(cells1.tank) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(cells1.tank$residuals) #plot histogram of residuals
boxplot(cells1.tank$residuals) #plot boxplot of residuals
plot(cells1.tank) #display residuals versus fitter, normal QQ plot, leverage plot

cells1.trt <- aov(cell.num ~Treatment, data=cells.1) #test the hypothesis feeding does not differ between treatments
cells1.trt.res <-anova(cells1.trt) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(cells1.trt$residuals) #plot histogram of residuals
boxplot(cells1.trt$residuals) #plot boxplot of residuals
plot(cells1.trt) #display residuals versus fitter, normal QQ plot, leverage plot

#Secondary Exposure: Quantifying feeding and testing for differences among treatments
cells.2 <- read.csv("RAnalysis/Data/Cell_Counts_Juvenile_Geoduck_Exp2.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
cells.2$cell.num <- rowMeans(cells.2[,c("Count1",  "Count2", "Count3")], na.rm = TRUE) #calculate average of counts
avg.cells.tank.2 <- do.call(data.frame,aggregate(cell.num ~ Tank, data = cells.2, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Tank
avg.cells.trt.2 <- do.call(data.frame,aggregate(cell.num ~ Treatment, data = cells.2, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Treatment
avg.cells.2 <- do.call(data.frame,aggregate(cell.num ~ Notes, data = cells.2, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Treatment
colnames(avg.cells.tank.2) <- c("Tank", "mean", "se") #rename columns 
colnames(avg.cells.trt.2) <- c("Treatment", "mean", "se") #rename columns 

cells2.tank <- aov(cell.num ~Tank, data=cells.2) #test the hypothesis feeding does not differ between tanks
cells2.tank.res <-anova(cells2.tank) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(cells2.tank$residuals) #plot histogram of residuals
boxplot(cells2.tank$residuals) #plot boxplot of residuals
plot(cells2.tank) #display residuals versus fitter, normal QQ plot, leverage plot

cells2.trt <- aov(cell.num ~Treatment, data=cells.2) #test the hypothesis feeding does not differ between treatments
cells2.trt.res <-anova(cells2.trt) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(cells2.trt$residuals) #plot histogram of residuals
boxplot(cells2.trt$residuals) #plot boxplot of residuals
plot(cells2.trt) #display residuals versus fitter, normal QQ plot, leverage plot
```


```{r Continuous pH}
#CONTINUOUS EXPERIMENTAL PH DATA 
#Avtech_pH_data.csv
pH <- read.csv("RAnalysis/Data/Avtech_pH_data.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
pH$Date.Time <-as.POSIXct(pH$Date.Time, format="%m/%d/%y %H:%M") #convert date format
pH$Date <- as.Date(pH$Date.Time) #convert Date only
pH$Time <- format(as.POSIXct(pH$Date.Time) ,format = "%H:%M:%S") #convert Time only

pH.low <- do.call(data.frame,aggregate(Low ~ Date, data = pH, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
pH.med <- do.call(data.frame,aggregate(Medium ~ Date, data = pH, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
pH.amb <- do.call(data.frame,aggregate(Ambient ~ Date, data = pH, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
pH.low$Treatment <- "Super Low" #Add treatment Information
colnames(pH.low) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
pH.med$Treatment <- "Low" #Add treatment Information
colnames(pH.med) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
pH.amb$Treatment <- "Ambient" #Add treatment Information
colnames(pH.amb) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
daily.pH <- rbind(pH.amb, pH.med, pH.low) #bind treatment data 
daily.pH #view data

# Plot daily averages of pH data for the complete experiment
All.pH <- ggplot(daily.pH, aes(x=Date, y=mean, group=Treatment)) + #set up plot information
  geom_errorbar(aes(ymin=daily.pH$mean-daily.pH$se, ymax=daily.pH$mean+daily.pH$se), colour="black", width=.1, position = position_dodge(width = 0.05)) + #add standard error bars about the mean
  geom_point(aes(shape=Treatment), size = 2, position = position_dodge(width = 0.05)) + #include points in the shape of the treatments
  annotate("text", x=as.Date("2016-03-25"), y=7.95, label = "No Data") + #add text to the graphic where data are missing
  annotate("text", x=as.Date("2016-06-15"), y=7.95, label = "No Data") + #add text to the graphic where data are missing
  xlab("Time") + #label x axis
  ylab("pH Total Scale") + # label y axis
  ylim(6.8,8.1) + # set y axis scale
  geom_vline(xintercept = 16899, linetype="dotted", color = "gray", size=1) + #add vertical line
  geom_vline(xintercept = 16928, linetype="dashed", color = "gray", size=1) + #add vertical line
  geom_vline(xintercept = 17013, linetype="dotted", color = "gray", size=1) + #add vertical line
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(), #Set the plot background
        legend.position=c(.65, .25), #set legend location
        legend.text = element_text(size = 8), #set the legend text size
        legend.key = element_blank(), #remove the legend background
        legend.title = element_text(size=8, face="bold")) + #set legend title attributes
  ggtitle("A) Experimental pH") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) #set title attributes
All.pH #view plot
```

```{r Continuous Temperature}
#CONTINUOUS EXPERIMENTAL TEMPERATURE DATA
#Avtech_temp_data.csv
temp <- read.csv("RAnalysis/Data/Avtech_temp_data.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
temp$Date.Time <-as.POSIXct(temp$Date.Time, format="%m/%d/%y %H:%M") #convert date format
temp$Date <- as.Date(temp$Date.Time) #convert Date only
temp$Time <- format(as.POSIXct(temp$Date.Time) ,format = "%H:%M:%S") #convert time only

temp.low <- do.call(data.frame,aggregate(Low ~ Date, data = temp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
temp.med <- do.call(data.frame,aggregate(Medium ~ Date, data = temp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
temp.amb <- do.call(data.frame,aggregate(Ambient ~ Date, data = temp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
temp.low$Treatment <- "Super Low" #Add treatment Information
colnames(temp.low) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
temp.med$Treatment <- "Low" #Add treatment Information
colnames(temp.med) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
temp.amb$Treatment <- "Ambient" #Add treatment Information
colnames(temp.amb) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
daily.temp <- rbind(temp.amb, temp.med, temp.low) #bind treatment data 
daily.temp #view data

All.temp <- ggplot(daily.temp, aes(x=Date, y=mean, group=Treatment)) + #set up plot information
  geom_errorbar(aes(ymin=daily.temp$mean-daily.temp$se, ymax=daily.temp$mean+daily.temp$se), colour="black", width=.1, position = position_dodge(width = 0.05)) + #add standard error bars about the mean
  geom_point(aes(shape=Treatment), size = 2, position = position_dodge(width = 0.05)) + #include points in the shape of the treatments
  annotate("text", x=as.Date("2016-05-25"), y=14, label = "No Data") + #add text to the graphic where data are missing
  xlab("Time") + #label x axis
  ylab("Temperature °C") + #label y axis
  ylim(0,20) + # set y axis scale
  geom_vline(xintercept = 16899, linetype="dotted", color = "gray", size=1) + #add vertical line
  geom_vline(xintercept = 16928, linetype="dashed", color = "gray", size=1) + #add vertical line
  geom_vline(xintercept = 17013, linetype="dotted", color = "gray", size=1) + #add vertical line
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(), #Set the plot background
        legend.position = "none") + #remove legend
  ggtitle("B) Experimental Temperature") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) #set title attributes
All.temp #view plot

FigureS1 <- arrangeGrob(All.pH, All.temp, ncol=1)
ggsave(file="RAnalysis/Figs/Fig.S1.pdf", FigureS1, width = 6, height = 8, units = c("in"))
```

```{r Discrete Seawater Chemistry}
#SEAWATER CHEMISTRY ANALYSIS FOR DISCRETE MEASUREMENTS

#pH Tris Calibration Curves
#Data to calculate conversion equations from mV to total scale using tris standard for pH probe
path <-("RAnalysis/Data/pH_Calibration_Files/") #set path to calibration file folder
file.names<-list.files(path = path, pattern = "csv$") #list all the file names with csv 
pH.cals <- data.frame(matrix(NA, nrow=length(file.names), ncol=4, dimnames=list(file.names,c("Date", "Intercept", "Slope","R2")))) #generate an empty 3 column dataframe with specific column names

for(i in 1:length(file.names)) { # for every file in list start at the first and run this following function
  Calib.Data <-read.table(file.path(path,file.names[i]), header=TRUE, sep=",", na.string="NA", as.is=TRUE) #reads in the data files
  model <-lm(mVTris ~ TTris, data=Calib.Data) #runs a linear regression of mV as a function of temperature
  coe <- coef(model) #extracts the coeffecients
  R <- summary(model)$r.squared #extracts the R2
  pH.cals[i,2:3] <- coe #inserts coef in the dataframe
  pH.cals[i,4] <- R #inserts R2 in the dataframe
  pH.cals[i,1] <- substr(file.names[i],1,8) #stores the file name in the Date column
}

colnames(pH.cals) <- c("Calib.Date",  "Intercept",  "Slope", "R2") #names the columns of the dataframe
pH.cals #view data

#Calculate Total pH
SW.chem <- read.csv("RAnalysis/Data/SW_Chem_Juvenile_Geoduck.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
SW.chem <- merge(pH.cals, SW.chem, by="Calib.Date") #merge pH calibrations with Seawater chemistry data
R <- 8.31447215 #gas constant in J mol-1 K-1 
F <-96485.339924 #Faraday constant in coulombs mol-1
mvTris <- SW.chem$Temperature*SW.chem$Slope+SW.chem$Intercept #calculate the mV of the tris standard using the temperature mv relationships in the measured standard curves 
STris<-27.5 #salinity of the Tris from local batch made for Manchester WA salinity conditions in Spring 2016. Batch date = 20160214
phTris<- (11911.08-18.2499*STris-0.039336*STris^2)*(1/(SW.chem$Temperature+273.15))-366.27059+ 0.53993607*STris+0.00016329*STris^2+(64.52243-0.084041*STris)*log(SW.chem$Temperature+273.15)-0.11149858*(SW.chem$Temperature+273.15) #calculate the pH of the tris (Dickson A. G., Sabine C. L. and Christian J. R., SOP 6a)
SW.chem$pH.Total<-phTris+(mvTris/1000-SW.chem$pH.MV/1000)/(R*(SW.chem$Temperature+273.15)*log(10)/F) #calculate the pH on the total scale (Dickson A. G., Sabine C. L. and Christian J. R., SOP 6a)
#SW.chem <- na.omit(SW.chem) #remove NAs

Trts.pH <- SW.chem %>%
  group_by(Treatment.Group, Exposure) %>%
  summarise(mean = mean(pH.Total,na.rm = TRUE), sem=std.error(pH.Total,na.rm = TRUE), n = n())
Trts.TA <- SW.chem %>%
  group_by(Treatment.Group, Exposure) %>%
  summarise(mean = mean(Total.Alkalinity,na.rm = TRUE),sem=std.error(Total.Alkalinity,na.rm = TRUE), n = n())
Trts.Temp<- SW.chem %>%
  group_by(Treatment.Group, Exposure) %>%
  summarise(mean = mean(Temperature,na.rm = TRUE),sem=std.error(Temperature,na.rm = TRUE), n = n())

TA.mod <- summary(aov(Total.Alkalinity ~ Treatment, data=SW.chem))
par(mfrow=c(1,3))
hist(TA.mod$residuals)
boxplot(TA.mod$residuals)
plot(TA.mod$fitted.values,TA.mod$residuals)
```

# ```{r}
#Prepare input file for seacarb

#apply mean TA to all values for the whole experiment


# #Calculate CO2 parameters using seacarb
# carb.output <- carb(flag=8, var1=SW.chem$pH.Total, var2=SW.chem$TA/1000000, S= SW.chem$Salinity, T=SW.chem$Temperature, P=0, Pt=0, Sit=0, pHscale="T", kf="pf", k1k2="l", ks="d") #calculate seawater chemistry parameters using seacarb
# carb.output$ALK <- carb.output$ALK*1000000 #convert to µmol kg-1
# carb.output$CO2 <- carb.output$CO2*1000000 #convert to µmol kg-1
# carb.output$HCO3 <- carb.output$HCO3*1000000 #convert to µmol kg-1
# carb.output$CO3 <- carb.output$CO3*1000000 #convert to µmol kg-1
# carb.output$DIC <- carb.output$DIC*1000000 #convert to µmol kg-1
# carb.output <- cbind(SW.chem$Measure.Date,  SW.chem$Exposure, SW.chem$Tank,  SW.chem$Treatment, carb.output) #combine the sample information with the seacarb output
# colnames(carb.output) <- c("Date", "Exposure", "Tank",  "Treatment",	"flag",	"Salinity",	"Temperature","Patm",	"Pressure",	"pH",	"CO2",	"fCO2", "pCO2",	"fCO2pot","pCO2pot","fCOinsitu", "pCOinsitu","HCO3",	"CO3",	"DIC", "TA",	"Aragonite.Sat", 	"Calcite.Sat") #Rename columns to describe contents
# carb.output <- subset(carb.output, select= c("Exposure","Date",  "Tank",  "Treatment",	"Salinity",	"Temperature",		"pH",	"CO2",	"pCO2",	"HCO3",	"CO3",	"DIC", "TA",	"Aragonite.Sat"))
# 
# #Calculate descriptive stats for seawater chemistry by Exposure
# ccarb <- melt(carb.output[,c(1,3,4,5:14)], id.vars=c("Exposure", "Treatment", "Tank")) #reshape data into long format
# Exp1 <-subset(ccarb, Exposure == "Exposure1") #separate out exposure 1 for all data
# Exp2 <-subset(ccarb, Exposure == "Exposure2") #separate out exposure 2 for all data
# 
# SWC.Tanks <- ddply(ccarb, c("Exposure", "Tank", "variable"), summarise, #apply functions to sewater chem data
#       N = length(na.omit(value)), #count the sample size removing NA
#       mean = mean(value), #calculate average 
#       sem = sd(value)/sqrt(N)) #calcualte the standard error of the mean
# 
# #Test for tank and treatment differences in Temperature and Total Alkalinity in Exposure 1
# Exp1.Temp <-subset(Exp1, variable=="Temperature") #separate out exposure 1 for all data
# temp1.tank <- aov(value ~Tank, data=Exp1.Temp) #test the hypothesis there is no difference in temperature between tanks
# temp1.tank.res <-anova(temp1.tank) #view results
# par(mfrow=c(3,2)) #set plotting configuration
# par(mar=c(1,1,1,1)) #set margins for plots
# hist(temp1.tank$residuals) #plot histogram of residuals
# boxplot(temp1.tank$residuals) #plot boxplot of residuals
# plot(temp1.tank) #display residuals versus fitter, normal QQ plot, leverage plot
# 
# temp1.trt <- aov(value ~Treatment, data=Exp1.Temp) #test the hypothesis there is no difference in temperature between treatments
# temp1.trt.res <- anova(temp1.trt) #statistical results
# par(mfrow=c(3,2)) #set plotting configuration
# par(mar=c(1,1,1,1)) #set margins for plots
# hist(temp1.trt$residuals) #plot histogram of residuals
# boxplot(temp1.trt$residuals) #plot boxplot of residuals
# plot(temp1.trt) #display residuals versus fitter, normal QQ plot, leverage plot
# 
# Exp1.TA <-subset(Exp1, variable=="TA") #separate out exposure 1 for all data
# TA1.tank <- aov(value ~Tank, data=Exp1.TA) #test the hypothesis there is no difference in total alkalinity between tanks
# TA1.tank.res <- anova(temp1.trt) #statistical results
# par(mfrow=c(3,2)) #set plotting configuration
# par(mar=c(1,1,1,1)) #set margins for plots
# hist(TA1.tank$residuals) #plot histogram of residuals
# boxplot(TA1.tank$residuals) #plot boxplot of residuals
# plot(TA1.tank) #display residuals versus fitter, normal QQ plot, leverage plot
# 
# TA1.trt <- aov(value ~Treatment, data=Exp1.TA) #test the hypothesis there is no difference in total alkalinity between treatments
# TA1.trt.res <- anova(temp1.trt) #statistical results
# par(mfrow=c(3,2)) #set plotting configuration
# par(mar=c(1,1,1,1)) #set margins for plots
# hist(TA1.trt$residuals) #plot histogram of residuals
# boxplot(TA1.trt$residuals) #plot boxplot of residuals
# plot(TA1.trt) #display residuals versus fitter, normal QQ plot, leverage plot
# 
# #Test for tank and treatment differences in Temperature and Total Alkalinity in Exposure 2
# Exp2.Temp <-subset(Exp2, variable=="Temperature") #separate out exposure 2 for all data
# temp2.tank <- aov(value ~Tank, data=Exp2.Temp) #test the hypothesis there is no difference in temperature between tanks
# temp2.tank.res <-anova(temp2.tank) #view results
# par(mfrow=c(3,2)) #set plotting configuration
# par(mar=c(1,1,1,1)) #set margins for plots
# hist(temp2.tank$residuals) #plot histogram of residuals
# boxplot(temp2.tank$residuals) #plot boxplot of residuals
# plot(temp2.tank) #display residuals versus fitter, normal QQ plot, leverage plot
# 
# temp2.trt <- aov(value ~Treatment, data=Exp2.Temp) #test the hypothesis there is no difference in temperature between treatments
# temp2.trt.res <- anova(temp2.trt) #statistical results
# par(mfrow=c(3,2)) #set plotting configuration
# par(mar=c(1,1,1,1)) #set margins for plots
# hist(temp2.trt$residuals) #plot histogram of residuals
# boxplot(temp2.trt$residuals) #plot boxplot of residuals
# plot(temp2.trt) #display residuals versus fitter, normal QQ plot, leverage plot
# 
# Exp2.TA <-subset(Exp2, variable=="TA") #separate out exposure 2 for all data
# TA2.tank <- aov(value ~Tank, data=Exp2.TA) #test the hypothesis there is no difference in total alkalinity between tanks
# TA2.tank.res <- anova(temp2.trt) #statistical results
# par(mfrow=c(3,2)) #set plotting configuration
# par(mar=c(1,1,1,1)) #set margins for plots
# hist(TA2.tank$residuals) #plot histogram of residuals
# boxplot(TA2.tank$residuals) #plot boxplot of residuals
# plot(TA2.tank) #display residuals versus fitter, normal QQ plot, leverage plot
# 
# TA2.trt <- aov(value ~Treatment, data=Exp2.TA) #test the hypothesis there is no difference in total alkalinity between treatments
# TA2.trt.res <- anova(temp2.trt) #statistical results
# par(mfrow=c(3,2)) #set plotting configuration
# par(mar=c(1,1,1,1)) #set margins for plots
# hist(TA2.trt$residuals) #plot histogram of residuals
# boxplot(TA2.trt$residuals) #plot boxplot of residuals
# plot(TA2.trt) #display residuals versus fitter, normal QQ plot, leverage plot
# 
# #Calculate descriptive stats by Treatment
# SWC.Treatments <- ddply(ccarb, c("Exposure", "Treatment", "variable"), summarise,
#       N = length(na.omit(value)), #count the sample size removing NA
#       mean = mean(value), #calculate average 
#       sem = sd(value)/sqrt(N)) #calcualte the standard error of the mean
# 
# Exposure1 <-subset(SWC.Treatments, Exposure == "Exposure1") #separate out exposure 1
# Exposure2 <-subset(SWC.Treatments, Exposure == "Exposure2") #separate out exposure 2
# Exposure1.long <- reshape(Exposure1, idvar="Treatment", direction="wide", timevar = "variable", drop = c("Exposure", "N")) #reshape data format for table layout
# Exposure2.long <- reshape(Exposure2, idvar="Treatment", direction="wide", timevar = "variable", drop = c("Exposure", "N")) #reshape data format for table layout
# write.table (Exposure1.long, file="/Users/hputnam/MyProjects/Geoduck_Epi/project_juvenile_geoduck_oa/RAnalysis/Output/Seawater_chemistry_table_Output_Seed_exposure1.csv", sep=",", row.names = FALSE) #save data to output file
# write.table (Exposure2.long, file="/Users/hputnam/MyProjects/Geoduck_Epi/project_juvenile_geoduck_oa/RAnalysis/Output/Seawater_chemistry_table_Output_Seed_exposure2.csv", sep=",", row.names = FALSE) #save data to output file
# 
# ```

```{r Shell Size}
##### Shell Size
##### JUVENILE GEODUCK SHELL SIZE #####

seed.size <- read.csv("RAnalysis/Data/Size_Juvenile_Geoduck.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA

#Initial Exposure Experimental Component
Initial <- subset(seed.size, Component=="Exp1", select = Date:Area) #subset dataframe to exposure 1 data
Initial$Ratio <- Initial$Length/Initial$Width #calculate the length:width of the shell

#calculating shell area means for Exposure 1 normalizations
E1.norms <- aggregate(Area ~ Day*Treatment, data=Initial, mean) #calculate mean area by Day and Treatment
E1.norm.area.amb <- subset(E1.norms, Day=="Day1" & Treatment=="Ambient", select = Area) #subset mean area for day 1 ambient condition
E1.norm.area.med <- subset(E1.norms, Day=="Day1" & Treatment=="Medium", select = Area) #subset mean area for day 1 medium condition
E1.norm.area.low <- subset(E1.norms, Day=="Day1" & Treatment=="Low", select = Area) #subset mean area for day 1 low condition

E1.norms <- function(x) {  #write function
  if(x == "Ambient") y <- E1.norm.area.amb #if Treatment equals Ambient assign day 1 Ambient mean as normalization factor
  if(x == "Medium") y <- E1.norm.area.med #if Treatment equals Medium assign day 1 Medium mean as normalization factor
  if(x == "Low") y <- E1.norm.area.low #if Treatment equals Low assign day 1 Low mean as normalization factor
  return(y) #return result
}

Initial$A.norm <- as.numeric(sapply(Initial$Treatment,E1.norms)) #add a column with the normalization values 
Initial$A.rel <- Initial$Area/Initial$A.norm #normalize the area to be relative size
Init.area <- do.call(data.frame,aggregate(Area ~ Day*Treatment, data = Initial, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
Init.Len <- do.call(data.frame,aggregate(Length ~ Day*Treatment, data = Initial, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
Init.Wid <- do.call(data.frame,aggregate(Width ~ Day*Treatment, data = Initial, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
Init.Ratio <- do.call(data.frame,aggregate(Ratio ~ Day*Treatment, data = Initial, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
Init.ARel <- do.call(data.frame,aggregate(A.rel ~ Day*Treatment, data = Initial, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
Init.Shell.size <- cbind(Init.area, Init.Len$Length.mean, Init.Len$Length.se, Init.Wid$Width.mean, Init.Wid$Width.se, Init.Ratio$Ratio.mean, Init.Ratio$Ratio.se, Init.ARel$A.rel.mean, Init.ARel$A.rel.se) #bind columns into dataframe
colnames(Init.Shell.size) <- c("Day", "Treatment", "avg.area", "se.area","avg.len", "se.len", "avg.wid", "se.wid","avg.ratio", "se.ratio","avg.Arel", "se.Arel") #rename columns

# Significance testing for initial exposure
Init <- aov(log10(A.rel) ~ Treatment*Day, data=Initial) #test the hypothesis relative size does not differ between treatments and time
Init.res <-anova(Init) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(Init$residuals) #plot histogram of residuals
boxplot(Init$residuals) #plot boxplot of residuals
plot(Init) #display residuals versus fitter, normal QQ plot, leverage plot

#PostHoc Tukey adjustment comparison of least-squares means
E1.lsm <- lsmeans(Init, ~ Treatment*Day, adjust="tukey") #compute least-squares means for Treatment*Day from ANOVA model
E1.pairs <- multcomp::cld(E1.lsm, alpha=.05, Letters=letters) #list pairwise tests and letter display
E1.pairs #view results

#Exposure 1 Plotting
Init.Shell.size$Day.Num <- c(1,10,23,1,10,23,1,10,23) #set days as numeric for plotting


Fig.Exp1.size <- ggplot(Init.Shell.size, aes(x=Day.Num, y=avg.Arel, group=Treatment)) + 
  geom_errorbar(aes(ymin=Init.Shell.size$avg.Arel-Init.Shell.size$se.Arel, ymax=Init.Shell.size$avg.Arel+Init.Shell.size$se.Arel), colour="black", width=.1, position = position_dodge(width = 0.6)) +
  geom_line(aes(linetype=Treatment, colour = Treatment), size = 0.5, position = position_dodge(width = 0.6)) + 
  scale_color_manual(values=c("blue","red", "mediumvioletred")) +
  geom_point(aes(shape=Treatment, colour=Treatment), size = 3, position = position_dodge(width = 0.6)) +
  scale_fill_manual(values=c("blue","red", "mediumvioletred")) +
  scale_x_continuous(breaks=seq(0,160,10)) +
  annotate("text", x=9, y=0.73, label = "a", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=1, y=1.2, label = "abc", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=11, y=1, label = "ab", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=c(9,22.5), y=c(1.2,0.95), label = "b", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=22.5, y=1.25, label = "bc", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=22, y=1.6, label = "c", size = 3) + #add text to the graphic for posthoc letters
  xlab("Days") +
  ylab(expression(paste("Relative Size"))) +
  ylim(0.5,3.5) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.key = element_blank(),  #remove legend background
        legend.position=c(.5, .7)) + #set legend location
  ggtitle("A) Initial Exposure") +
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0))
Fig.Exp1.size

#Common Garden Experimental Component
CG.data <- subset(seed.size, Component=="Common.Garden", select = Date:Area) #subset dataframe to common garden data
CG.data$Ratio <- CG.data$Length/CG.data$Width #calculate the length:width of the shell

#calculating shell area means for Common Garden normalizations
CG.norm.area.amb <- subset(Init.Shell.size, Day=="Day23" & Treatment=="Ambient", select = avg.area) #subset mean area for day 23 ambient condition
CG.norm.area.med <- subset(Init.Shell.size, Day=="Day23" & Treatment=="Medium", select = avg.area) #subset mean area for day 23 medium condition
CG.norm.area.low <- subset(Init.Shell.size, Day=="Day23" & Treatment=="Low", select = avg.area) #subset mean area for day 23 high condition

CG.norms <- function(x) { 
  if(x == "Ambient") y <- CG.norm.area.amb #if Treatment equals Ambient assign day 23 Ambient mean as normalization factor
  if(x == "Medium") y <- CG.norm.area.med #if Treatment equals Medium assign day 23 Medium mean as normalization factor
  if(x == "Low") y <- CG.norm.area.low #if Treatment equals Low assign day 23 Low mean as normalization factor
  return(y)
}

CG.data$A.norm <- as.numeric(sapply(CG.data$Treatment,CG.norms)) #add a column with the normalization values
CG.data$A.rel <- CG.data$Area/CG.data$A.norm #normalize the area to be relative size
CG.area <- do.call(data.frame,aggregate(Area ~ Day*Treatment, data = CG.data, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
CG.Len <- do.call(data.frame,aggregate(Length ~ Day*Treatment, data = CG.data, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
CG.Wid <- do.call(data.frame,aggregate(Width ~ Day*Treatment, data = CG.data, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
CG.Ratio <- do.call(data.frame,aggregate(Ratio ~ Day*Treatment, data = CG.data, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
CG.ARel <- do.call(data.frame,aggregate(A.rel ~ Day*Treatment, data = CG.data, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
CG.Shell.size <- cbind(CG.area, CG.Len$Length.mean, CG.Len$Length.se, CG.Wid$Width.mean, CG.Wid$Width.se, CG.Ratio$Ratio.mean, CG.Ratio$Ratio.se, CG.ARel$A.rel.mean, CG.ARel$A.rel.se) #bind columns into dataframe
colnames(CG.Shell.size) <- c("Day", "Treatment", "avg.area", "se.area","avg.len", "se.len", "avg.wid", "se.wid","avg.ratio", "se.ratio","avg.Arel", "se.Arel") #rename columns

# Significance testing for common garden
CG <- aov(log10(A.rel) ~ Treatment*Day, data=CG.data)
CG.res <- anova(CG)
par(mfrow=c(3,2))
par(mar=c(1,1,1,1))
hist(CG$residuals)
boxplot(CG$residuals)
plot(CG)

#PostHoc Tukey adjustment comparison of least-squares means
CG.lsm <- lsmeans(CG, ~ Treatment*Day, adjust="tukey") #compute least-squares means for Treatment*Day from ANOVA model
CG.pairs <- multcomp::cld(CG.lsm, alpha=.05, Letters=letters) #list pairwise tests and letter display
CG.pairs #view results

#Common Garden Plotting
CG.Shell.size$Day.Num <- c(135,51,135,51,135,51) #set days as numeric for plotting

Fig.CG.size <- ggplot(CG.Shell.size, aes(x=Day.Num, y=avg.Arel, group=Treatment)) + 
  geom_errorbar(aes(ymin=CG.Shell.size$avg.Arel-CG.Shell.size$se.Arel, ymax=CG.Shell.size$avg.Arel+CG.Shell.size$se.Arel), colour="black", width=.1, position = position_dodge(width = 0.6)) +
  geom_line(aes(linetype=Treatment, colour=Treatment), size = 0.5, position = position_dodge(width = 0.6)) +
  scale_color_manual(values=c("blue","red", "mediumvioletred")) +
  geom_point(aes(shape=Treatment, colour=Treatment), size = 3, position = position_dodge(width = 0.6)) +
  scale_fill_manual(values=c("blue","red", "mediumvioletred")) +
  scale_x_continuous(breaks=seq(0,160,10)) +
  annotate("text", x=43, y=1.85, label = "a", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=132, y=2.15, label = "b", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=c(43,43), y=c(2.45,2.2), label = "ab", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=c(141,141), y=c(3.15,3.4), label = "c", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=35, y=0.5, label = ".", size = 1) + #add text to the graphic for posthoc letters
  xlab("Days") +
  ylab(expression(paste("Relative Size"))) +
  ylim(0.5,3.5) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position='none') + #remove legend background
  ggtitle("B) Common Garden") +
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0))
Fig.CG.size

#Exposure 2 Experimental Component
ReExp <- subset(seed.size, Day=="Day145", select = Date:Area) #subset dataframe to exposure 2 data
ReExp$Ratio <- ReExp$Length/ReExp$Width #calculate the length:width of the shell

#calculating mean for Exposure 2 normalizations
E2.norm.area.amb <- subset(CG.Shell.size, Day=="Day135" & Treatment=="Ambient", select = avg.area) #subset mean area for day 135 ambient condition
E2.norm.area.med <- subset(CG.Shell.size, Day=="Day135" & Treatment=="Medium", select = avg.area) #subset mean area for day 135 medium condition
E2.norm.area.low<- subset(CG.Shell.size, Day=="Day135" & Treatment=="Low", select = avg.area) #subset mean area for day 135 low condition

E2.norms <- function(x) { 
  if(x == "Ambient") y <- E2.norm.area.amb #if Treatment equals Ambient assign day 135 Ambient mean as normalization factor
  if(x == "Medium") y <- E2.norm.area.med #if Treatment equals Ambient assign day 135 Medium mean as normalization factor
  if(x == "Low") y <- E2.norm.area.low #if Treatment equals Ambient assign day 135 Low mean as normalization factor
  return(y)
}

ReExp$A.norm <- as.numeric(sapply(ReExp$Treatment,E2.norms)) #add a column with the normalization values
ReExp$A.rel <- ReExp$Area/ReExp$A.norm #normalize the area to be relative size
ReExp.area <- do.call(data.frame,aggregate(Area ~ Treatment*Secondary*Day, data = ReExp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
ReExp.Len <- do.call(data.frame,aggregate(Length ~ Treatment*Secondary*Day, data = ReExp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
ReExp.Wid <- do.call(data.frame,aggregate(Width ~ Treatment*Secondary*Day, data = ReExp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
ReExp.Ratio <- do.call(data.frame,aggregate(Ratio ~ Treatment*Secondary*Day, data = ReExp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
ReExp.ARel <- do.call(data.frame,aggregate(A.rel ~ Treatment*Secondary*Day, data = ReExp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem
ReExp.Shell.size <- cbind(ReExp.area, ReExp.Len$Length.mean, ReExp.Len$Length.se, ReExp.Wid$Width.mean, ReExp.Wid$Width.se, ReExp.Ratio$Ratio.mean, ReExp.Ratio$Ratio.se, ReExp.ARel$A.rel.mean, ReExp.ARel$A.rel.se) #bind columns into dataframe
colnames(ReExp.Shell.size) <- c("Treatment", "Secondary", "Day", "avg.area", "se.area","avg.len", "se.len", "avg.wid", "se.wid","avg.ratio", "se.ratio","avg.Arel", "se.Arel") #rename columns

# Significance testing for secondary exposure
Rexps <- aov(log10(A.rel) ~ Treatment*Secondary, data=ReExp)
Rexp.res <- anova(Rexps)
Rexp.res
par(mfrow=c(3,2))
par(mar=c(1,1,1,1))
hist(Rexps$residuals)
boxplot(Rexps$residuals)
plot(Rexps)

#PostHoc
Day145 <- subset(ReExp.Shell.size, Day=="Day145", select = Treatment:se.Arel)
#Day158 <- subset(ReExp.Shell.size, Day=="Day158", select = Treatment:se.Arel)

E2.lsm <- lsmeans(Rexps, ~ Treatment*Secondary, adjust="tukey") #compute least-squares means for Treatment*Day from ANOVA model
E2.pairs <- multcomp::cld(E2.lsm, alpha=.05, Letters=letters) #list pairwise tests and letter display
E2.pairs #view results

#Exposure2 Plotting
Fig.Exp2.D10.size <- ggplot(Day145, aes(x=Secondary, y=avg.Arel, group=Treatment)) + 
  geom_errorbar(aes(ymin=Day145$avg.Arel-Day145$se.Arel, ymax=Day145$avg.Arel+Day145$se.Arel), colour="black", width=.1, position = position_dodge(width = 0.05)) +
  geom_line(aes(linetype=Treatment, colour=Treatment), size = 0.5, position = position_dodge(width = 0.05)) +
  scale_color_manual(values=c("blue","red", "mediumvioletred")) +
  geom_point(aes(shape=Treatment, colour=Treatment), size = 3, position = position_dodge(width = 0.05)) +
  scale_fill_manual(values=c("blue","red", "mediumvioletred")) +
  annotate("text", x=0.85, y=1.3, label = "b", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=0.85, y=0.88, label = "a", size = 3) + #add text to the graphic for posthoc letters
  annotate("text", x=c(0.85,2.2,2.2,2.25), y=c(0.94,0.94, 0.88,0.97), label = "ab", size = 3) + #add text to the graphic for posthoc letters
  xlab("Secondary Treatment") +
  ylab("Relative Size") +
  ylim(0.5,1.5) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(), #Set the plot background
        legend.position='none') + #remove legend background
  ggtitle("C) Secondary Exposure") +
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0))

Fig.Exp2.D10.size

##### CAPTURE STATISTICAL OUTPUT AND FIGURES TO FILE #####
#Capture Figures to File
Figure1.Size <- arrangeGrob(Fig.Exp1.size,Fig.CG.size, Fig.Exp2.D10.size, ncol=3)
ggsave(file="RAnalysis/Output/Geoduck_Size.pdf", Figure1.Size, width = 8, height = 6, units = c("in"))
ggsave(file="RAnalysis/Figs/Geoduck_Size.png", Figure1.Size, width = 8, height = 6, units = c("in"))

```

# Load Genomic Features and Sample Info

```{r}
#load sample information
sample.info <- read.csv("RAnalysis/Data/Sample.Info.csv", header=T, sep=",", na.string="NA", stringsAsFactors = F) #read in info
samp <- sample.info$Sample.ID # set sample info
samp <- gsub("[_]", "-", samp) #remove extra characters

#load genes gff
Genes <- read.csv("RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3", header=F, sep="\t", na.string="NA", skip=3, stringsAsFactors = F) #read in data file
Genes <- Genes[,c(1,4,5,9)] #select desired columns only
colnames(Genes) <- c("scaffold", "start", "stop", "gene") #rename columns
Genes$gene <- gsub(";.*","",Genes$gene) #remove extra characters
Genes$gene <- gsub("ID=","",Genes$gene) #remove extra characters

#Load annotation file
Annot <- read.csv("RAnalysis/Data/Genome/Panopea-generosa-genes-annotations.tab", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = F, skip=0)
colnames(Annot) <- c("gene","scaffold", "start", "end", "uniprot.id", "uniprot_taxa.id", "Uniprot.Name", "GO.IDs", "GO.Descript")
GO.data <- Annot[,c(1,8)] #select names and GOs
splitted <- strsplit(as.character(GO.data$GO.IDs), "; ") #split into multiple GO ids
Gene.GO.IDs <- data.frame(v1 = rep.int(GO.data$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 
colnames(Gene.GO.IDs) <- c("gene", "GO.IDs") #rename columns
Gene.GO.IDs$GO.IDs <- as.character(Gene.GO.IDs$GO.IDs)
Gene.GO.IDs$GO.IDs[is.na(Gene.GO.IDs$GO.IDs)] <- "unknown" #replace NA with unknown
GO.IDs <-unique(Gene.GO.IDs$GO.IDs)

```

# Filter and Concatenate Datasets
#https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/032120-fds/

```{bash}
#Time dataset
#mkdir RAnalysis/Data/OSF/Time 


multiIntersectBed -i \
RAnalysis/Data/OSF/Time/EPI-41_S38_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-42_S39_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-43_S40_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-44_S41_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-119_S31_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-120_S32_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-135_S35_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-136_S36_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-151_S2_L002_5x.tab \
RAnalysis/Data/OSF/Time/EPI-152_S3_L002_5x.tab \
RAnalysis/Data/OSF/Time/EPI-153_S4_L002_5x.tab \
RAnalysis/Data/OSF/Time/EPI-154_S5_L002_5x.tab \
RAnalysis/Data/OSF/Time/EPI-181_S16_L003_5x.tab \
RAnalysis/Data/OSF/Time/EPI-182_S17_L003_5x.tab \
RAnalysis/Data/OSF/Time/EPI-184_S18_L003_5x.tab \
RAnalysis/Data/OSF/Time/EPI-185_S19_L003_5x.tab \
> RAnalysis/Data/OSF/Time/time.5x.bed

cat RAnalysis/Data/OSF/Time/time.5x.bed | awk '$4 ==16' > RAnalysis/Data/OSF/Time/time.filtered.16.5x.bed 

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file that ends in posOnly
#b: annotated gene list
#Save output in a new file that has the same base name and ends in -Annotated.txt
for f in RAnalysis/Data/OSF/Time/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in RAnalysis/Data/OSF/Time/*_gene
do
  intersectBed \
  -a ${f} \
  -b RAnalysis/Data/OSF/Time/time.filtered.16.5x.bed  \
  > ${f}_CpG_time.bed
done

wc -l RAnalysis/Data/OSF/Time/*_CpG_time.bed
```

```{bash}
# Day 10 Data Set
#mkdir RAnalysis/Data/OSF/D10 

multiIntersectBed -i \
RAnalysis/Data/OSF/D10/EPI-103_S27_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-104_S28_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-111_S29_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-113_S30_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-119_S31_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-120_S32_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-127_S33_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-128_S34_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-135_S35_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-136_S36_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-143_S37_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-145_S38_L005_5x.tab \
> RAnalysis/Data/OSF/D10/D10.5x.bed

cat RAnalysis/Data/OSF/D10/D10.5x.bed | awk '$4 ==12' > RAnalysis/Data/OSF/D10/D10.filtered.12.5x.bed 


#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in RAnalysis/Data/OSF/D10/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in RAnalysis/Data/OSF/D10/*_gene
do
  intersectBed \
  -a ${f} \
  -b RAnalysis/Data/OSF/D10/D10.filtered.12.5x.bed   \
  > ${f}_CpG_D10.bed
done

wc -l RAnalysis/Data/OSF/D10/*_CpG_D10.bed
```

```{bash}
# Day 135 Data

#mkdir RAnalysis/Data/OSF/D135 

multiIntersectBed -i \
RAnalysis/Data/OSF/D135/EPI-151_S2_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-152_S3_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-153_S4_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-154_S5_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-159_S6_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-160_S7_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-161_S8_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-162_S9_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-167_S10_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-168_S11_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-169_S12_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-170_S13_L002_5x.tab \
> RAnalysis/Data/OSF/D135/D135.5x.bed

cat RAnalysis/Data/OSF/D135/D135.5x.bed | awk '$4 ==12' > RAnalysis/Data/OSF/D135/D135.filtered.12.5x.bed 

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in RAnalysis/Data/OSF/D135/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in RAnalysis/Data/OSF/D135/*_gene
do
  intersectBed \
  -a ${f} \
  -b RAnalysis/Data/OSF/D135/D135.filtered.12.5x.bed   \
  > ${f}_CpG_D135.bed
done

wc -l RAnalysis/Data/OSF/D135/*_CpG_D135.bed
```

```{bash}
# Day 145 Data

#mkdir RAnalysis/Data/OSF/D145 

multiIntersectBed -i \
RAnalysis/Data/OSF/D145/EPI-175_S14_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-176_S15_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-181_S16_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-182_S17_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-184_S18_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-185_S19_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-187_S20_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-188_S21_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-193_S22_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-194_S23_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-199_S24_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-200_S25_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-205_S26_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-206_S27_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-208_S28_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-209_S29_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-214_S30_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-215_S31_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-220_S32_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-221_S33_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-226_S34_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-227_S35_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-229_S36_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-230_S37_L004_5x.tab \
> RAnalysis/Data/OSF/D145/D145.5x.bed

cat RAnalysis/Data/OSF/D145/D145.5x.bed | awk '$4 ==24' > RAnalysis/Data/OSF/D145/D145.filtered.24.5x.bed 

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in RAnalysis/Data/OSF/D145/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in RAnalysis/Data/OSF/D145/*_gene
do
  intersectBed \
  -a ${f} \
  -b RAnalysis/Data/OSF/D145/D145.filtered.24.5x.bed   \
  > ${f}_CpG_D145.bed
done

wc -l RAnalysis/Data/OSF/D145/*_CpG_D145.bed
```

# Load in methylation counts

```{r}
# Time
meth.data.T <- list.files(path = "RAnalysis/Data/OSF/Time", pattern = "time.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = FALSE) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.T) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.T$gene <- gsub(";.*","",meth.data.T$gene) #remove extra characters
meth.data.T$gene <- gsub("ID=","",meth.data.T$gene) #remove extra characters
meth.data.T$Sample.ID <- gsub("RAnalysis/Data/OSF/Time/","",meth.data.T$Sample.ID) #remove extra characters
meth.data.T$Sample.ID <- gsub("_.*","",meth.data.T$Sample.ID) #remove extra characters 
meth.data.T$Sample.ID <- gsub("-","_",meth.data.T$Sample.ID) #remove extra characters

MD.Time <- merge(meth.data.T,sample.info, by="Sample.ID")
```

```{r}
#Day 10
meth.data.D10 <- list.files(path = "RAnalysis/Data/OSF/D10", pattern = "D10.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = FALSE) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)

colnames(meth.data.D10) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D10$gene <- gsub(";.*","",meth.data.D10$gene) #remove extra characters
meth.data.D10$gene <- gsub("ID=","",meth.data.D10$gene) #remove extra characters
meth.data.D10$Sample.ID <- gsub("RAnalysis/Data/OSF/D10/","",meth.data.D10$Sample.ID) #remove extra characters
meth.data.D10$Sample.ID <- gsub("_.*","",meth.data.D10$Sample.ID) #remove extra characters 
meth.data.D10$Sample.ID <- gsub("-","_",meth.data.D10$Sample.ID) #remove extra characters

MD.D10 <- merge(meth.data.D10, sample.info, by="Sample.ID")
```

```{r}
#Day 135
meth.data.D135 <- list.files(path = "RAnalysis/Data/OSF/D135", pattern = "D135.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = FALSE) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.D135) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D135$gene <- gsub(";.*","",meth.data.D135$gene) #remove extra characters
meth.data.D135$gene <- gsub("ID=","",meth.data.D135$gene) #remove extra characters
meth.data.D135$Sample.ID <- gsub("RAnalysis/Data/OSF/D135/","",meth.data.D135$Sample.ID) #remove extra characters
meth.data.D135$Sample.ID <- gsub("_.*","",meth.data.D135$Sample.ID) #remove extra characters 
meth.data.D135$Sample.ID <- gsub("-","_",meth.data.D135$Sample.ID) #remove extra characters

MD.D135 <- merge(meth.data.D135, sample.info, by="Sample.ID")
```

```{r}
#Day 145
meth.data.D145 <- list.files(path = "RAnalysis/Data/OSF/D145", pattern = "D145.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = FALSE) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.D145) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D145$gene <- gsub(";.*","", meth.data.D145$gene) #remove extra characters
meth.data.D145$gene <- gsub("ID=","",meth.data.D145$gene) #remove extra characters
meth.data.D145$Sample.ID <- gsub("RAnalysis/Data/OSF/D145/","",meth.data.D145$Sample.ID) #remove extra characters
meth.data.D145$Sample.ID <- gsub("_.*","",meth.data.D145$Sample.ID) #remove extra characters 
meth.data.D145$Sample.ID <- gsub("-","_",meth.data.D145$Sample.ID) #remove extra characters

MD.D145 <- merge(meth.data.D145, sample.info, by="Sample.ID")
```

```{r}
#generate list of all genes identified and background list for GO enrichment analysis
MD.ALL <- rbind(MD.Time, MD.D10, MD.D135, MD.D145)
MD.ALL.Genes <- unique(MD.ALL$gene)
write.csv(MD.ALL.Genes, file = "RAnalysis/Output/MD.ALL.Genes_For_GO_Background.csv")
```

# Testing for Differentially Methylated Genes

```{r}
# Comparison of Ambient samples through Time
# Binomial GLM to test for differentially methylated genes

sub_meth_table  <- MD.Time
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to stored results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ TimePoint, 
             data=sub_meth_table1, family=binomial)
  a <- anova(fit, test="Chisq")
  
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3], #uncomment if you want to include position of CpG within a gene
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4], #uncomment if you want to include position of CpG within a gene interaction with treatment
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}
# An error will be generated here for contrasts. 
#This potential for contrasts (interactions) is included in the case one wants to examine the role of position of CpG within a gene
#Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : contrasts can be applied only to factors with 2 or more levels
#Continuing the analysis from results line will generate the results in the absence of the contrast (interaction).

results[is.na(results)] <- 0

results$adj.pval.TimePoint <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH') #uncomment if you want to include position of CpG within a gene
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH') #uncomment if you want to include position of CpG within a gene interaction with treatment

Time.Amb.sig <-results
Time.Amb.sig <- Time.Amb.sig[,c(1,3)]
Time.Amb.sig <- Time.Amb.sig[order(Time.Amb.sig$adj.pval.TimePoint),]
Time.Amb.sig <- Time.Amb.sig[which(Time.Amb.sig$adj.pval.TimePoint<0.05), ]

#write.table(Time.Amb.sig, '/home/hputnam/MyPorjects/Geoduck_Meth/RAnalysis/Output/Time_sig.tsv', sep='\t', row.names=FALSE)

sub_meth_table$per.meth <- (sub_meth_table$meth/(sub_meth_table$meth+sub_meth_table$unmeth))*100 #calculate percent methylation

# Annotation of DMG under ambient conditions through time
Time.Amb.annot <- merge(Time.Amb.sig , Annot, by="gene", all.x=TRUE)
Time.Amb.annot <- Time.Amb.annot[order(Time.Amb.annot$adj.pval.TimePoint),]
Time.Amb.annot <- Time.Amb.annot[!duplicated(Time.Amb.annot$gene),]
write.table(Time.Amb.annot, 'RAnalysis/Output/Table_S1_Time_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Ambient samples through Time
##### GO enrichment of DMGs #####
ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(Time.Amb.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.Time <- merge(ALL, Gene.GO.IDs,by="gene", all=T) #merge GO, ID
GO.Time[is.na(GO.Time)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.Time$GO.IDs), "; ") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.Time$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

ALL.Time <-enriched.GO.05 %>% arrange(-numDEInCat) #arrange by order of num DE in cat largest to smallest
str(ALL.Time) #view 

MF.Time <- subset(ALL.Time, ontology=="MF")
MF.Time <- MF.Time[order(-MF.Time$numDEInCat),]
CC.Time <- subset(ALL.Time, ontology=="CC")
CC.Time <- CC.Time[order(-CC.Time$numDEInCat),]
BP.Time <- subset(ALL.Time, ontology=="BP")
BP.Time <- BP.Time[order(-BP.Time$numDEInCat),]
write.csv(ALL.Time , file = "RAnalysis/Output/ALL_Sig_Enriched_GO.05_Time.csv")
write.csv(MF.Time , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_Time.csv")
write.csv(CC.Time , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_Time.csv")
write.csv(BP.Time , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_Time.csv")
```


```{r}
# Comparison of Treatments at Day 10
# Binomial GLM to test for differentially methylated genes

meth_table  <- MD.D10
sub_meth_table <- meth_table[meth_table$Initial.Treatment == 'Ambient' | meth_table$Initial.Treatment == 'Low' | meth_table$Initial.Treatment == 'Super.Low', ]
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)
  
#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to stored results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment, 
             data=sub_meth_table1, family=binomial, maxit = 100)
  a <- anova(fit, test="Chisq")
  
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3], #uncomment if you want to include position of CpG within a gene
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4], #uncomment if you want to include position of CpG within a gene interaction with treatment
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}
# An error will be generated here for contrasts. 
#This potential for contrasts (interactions) is included in the case one wants to examine the role of position of CpG within a gene
#Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : contrasts can be applied only to factors with 2 or more levels
#Continuing the analysis from results line will generate the results in the absence of the contrast (interaction).

results[is.na(results)] <- 0

results$adj.pval.Initial.Treatment <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH') #uncomment if you want to include position of CpG within a gene
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH') #uncomment if you want to include position of CpG within a gene interaction with treatment

D10.sig <-results
D10.sig <- D10.sig[,c(1,3)]
D10.sig <- D10.sig[order(D10.sig$adj.pval.Initial.Treatment),]
D10.sig <- D10.sig[which(D10.sig$adj.pval.Initial.Treatment<0.05), ]

write.table(D10.sig, 'RAnalysis/Output/D10_sig.tsv', sep='\t', row.names=FALSE)

# Annotation of D10 DMG 
D10.sig.annot <- merge(D10.sig, Annot, by="gene", all.x=TRUE)
D10.sig.annot <- D10.sig.annot[order(D10.sig.annot$adj.pval.Initial.Treatment),]
D10.sig.annot <- D10.sig.annot[!duplicated(D10.sig.annot$gene),]
write.table(D10.sig.annot, 'RAnalysis/Output/Table_S2_D10_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 10
##### GO enrichment of DMGs #####
ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(D10.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D10 <- merge(ALL, Gene.GO.IDs,by="gene", all=T) #merge GO, ID
GO.D10[is.na(GO.D10)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D10$GO.IDs), "; ") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D10$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector <- as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

ALL.D10 <-enriched.GO.05 %>% arrange(-numDEInCat)
str(ALL.D10)

MF.D10 <- subset(ALL.D10, ontology=="MF")
MF.D10 <- MF.D10[order(-MF.D10$numDEInCat),]
CC.D10 <- subset(ALL.D10, ontology=="CC")
CC.D10 <- CC.D10[order(-CC.D10$numDEInCat),]
BP.D10 <- subset(ALL.D10, ontology=="BP")
BP.D10 <- BP.D10[order(-BP.D10$numDEInCat),]
write.csv(ALL.D10 , file = "RAnalysis/Output/ALL_Sig_Enriched_GO.05_D10.csv")
write.csv(MF.D10 , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_D10.csv")
write.csv(CC.D10 , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_D10.csv")
write.csv(BP.D10 , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_D10.csv")

```

```{r}
# Comparison of Treatments at Day 135
# Binomial GLM to test for differentially methylated genes

meth_table  <- MD.D135
sub_meth_table <- meth_table[meth_table$Initial.Treatment == 'Ambient' | meth_table$Initial.Treatment == 'Low' | meth_table$Initial.Treatment == 'Super.Low', ]

sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to store results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  #if(length(unique(sub_meth_table1$position)) >1){
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment, 
             data=sub_meth_table1, family=binomial)
  #} else {
  #  # fit glm model
  #  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment * position, 
  #             data=sub_meth_table1, family=binomial)
  #}
  #s <- step(fit, trace=0)
  a <- anova(fit, test="Chisq")
  a
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3],
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4],
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}
# An error will be generated here for contrasts. 
#This potential for contrasts (interactions) is included in the case one wants to examine the role of position of CpG within a gene
#Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : contrasts can be applied only to factors with 2 or more levels
#Continuing the analysis from results line will generate the results in the absence of the contrast (interaction).

results[is.na(results)] <- 0

results$adj.pval.treatment <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH')
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH')

D135.sig <-results
D135.sig <- D135.sig[,c(1,3)]
D135.sig <- D135.sig[order(D135.sig$adj.pval.treatment),]
D135.sig <- D135.sig[which(D135.sig$adj.pval.treatment<0.05), ]
write.table(D135.sig, 'RAnalysis/Output/D135_sig.tsv', sep='\t', row.names=FALSE)

# Annotation of D135 DMG 
D135.sig.annot <- merge(D135.sig, Annot, by="gene", all.x=TRUE)
D135.sig.annot <- D135.sig.annot[order(D135.sig.annot$adj.pval.treatment),]
D135.sig.annot <- D135.sig.annot[!duplicated(D135.sig.annot$gene),]
write.table(D135.sig.annot, 'RAnalysis/Output/Table_S3_D135_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 135
##### GO enrichment of DMGs #####
ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(D135.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D135 <- merge(ALL, Gene.GO.IDs,by="gene", all=T) #merge GO, ID
GO.D135[is.na(GO.D135)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D135$GO.IDs), "; ") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D135$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

ALL.D135 <-enriched.GO.05 %>% arrange(-numDEInCat)
str(ALL.D135)

MF.D135 <- subset(ALL.D135, ontology=="MF")
MF.D135 <- MF.D135[order(-MF.D135$numDEInCat),]
CC.D135 <- subset(ALL.D135, ontology=="CC")
CC.D135 <- CC.D135[order(-CC.D135$numDEInCat),]
BP.D135 <- subset(ALL.D135, ontology=="BP")
BP.D135 <- BP.D135[order(-BP.D135$numDEInCat),]
write.csv(ALL.D135 , file = "RAnalysis/Output/ALL_Sig_Enriched_GO.05_D135.csv")
write.csv(MF.D135 , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_D135.csv")
write.csv(CC.D135 , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_D135.csv")
write.csv(BP.D135 , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_D135.csv")

```

```{r}
# Comparison of Treatments at Day 145
# Binomial GLM to test for differentially methylated genes
meth_table <- MD.D145
sub_meth_table <- meth_table[meth_table$Initial.Treatment == "Ambient" | meth_table$Initial.Treatment == "Low" | meth_table$Initial.Treatment == "Super.Low"
                             | meth_table$Secondary.Treatment == "Ambient" | meth_table$Secondary.Treatment == "Low", ]
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to store results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
    
    #subset the dataframe gene by gene
    sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
    
    # fit glm position model
    fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment * Secondary.Treatment, 
               data=sub_meth_table1, family=binomial)

    a <- anova(fit, test="LRT")
    
    # capture summary stats to data frame
    df <- data.frame(gene = sub_meth_table1[1,7],
                     pval.treatment1 = a$`Pr(>Chi)`[2],
                     pval.treatment2 = a$`Pr(>Chi)`[3],
                     #position = a$`Pr(>Chi)`[2], 
                     pval.treatment1_x_treatment2 = a$`Pr(>Chi)`[4],
                     #pval.treatment1_x_position = a$`Pr(>Chi)`[2],
                     #pval.treatment2_x_position = a$`Pr(>Chi)`[2],
                     #pval.treatment1_x_pval.treatment2_x_position = a$`Pr(>Chi)`[2],
                     stringsAsFactors = F)

    # bind rows of temporary data frame to the results data frame
    results <- rbind(results, df)
    
  }
# An error will be generated here for contrasts. 
#This potential for contrasts (interactions) is included in the case one wants to examine the role of position of CpG within a gene
#Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : contrasts can be applied only to factors with 2 or more levels
#Continuing the analysis from results line will generate the results in the absence of the contrast (interaction).

results[is.na(results)] <- 0

results$adj.pval.treatment1 <- p.adjust(results$pval.treatment1, method='BH')
results$adj.pval.treatment2 <- p.adjust(results$pval.treatment2, method='BH')
results$adj.pval.treatment1_x_treatment2 <- p.adjust(results$pval.treatment1_x_treatment2, method='BH')
#results$adj.pval.treatment1_x_position <- p.adjust(results$pval.treatment1_x_position, method='BH')
#results$adj.pval.treatment2_x_position <- p.adjust(results$pval.treatment2_x_position, method='BH')
#results$adj.pval.treatment1_x_pval.treatment2_x_position <- p.adjust(results$pval.treatment1_x_pval.treatment2_x_position, method='BH')

D145.sig <-results
D145.sig <- D145.sig[,c(1,5:7)]
D145.sig <- D145.sig[order(results$adj.pval.treatment1_x_treatment2),]
D145.sig <- D145.sig[which(D145.sig$adj.pval.treatment1_x_treatment2<0.05), ]
write.table(D145.sig, 'RAnalysis/Output/D145_sig.tsv', sep='\t', row.names=FALSE)

sub_meth_table$per.meth <- (sub_meth_table$meth/(sub_meth_table$meth+sub_meth_table$unmeth))*100
sub_meth_table.D145 <- sub_meth_table

# Annotation of Secondary Exposure Interaction DMG 
D145.sig.annot <- merge(D145.sig, Annot, by="gene", all.x=TRUE)
D145.sig.annot <- D145.sig.annot[order(D145.sig.annot$adj.pval.treatment1_x_treatment2),]
D145.sig.annot <- D145.sig.annot[!duplicated(D145.sig.annot$gene),]
write.table(D145.sig.annot, 'RAnalysis/Output/Table_S4_D145_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 145
##### GO enrichment of DMGs #####
ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(D145.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D145 <- merge(ALL, Gene.GO.IDs,by="gene", all=T) #merge GO, ID
GO.D145[is.na(GO.D145)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D145$GO.IDs), "; ") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D145$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

ALL.D145 <-enriched.GO.05 %>% arrange(-numDEInCat)
str(ALL.D145)
MF.D145 <- subset(ALL.D145, ontology=="MF")
MF.D145 <- MF.D145[order(-MF.D145$numDEInCat),]
CC.D145 <- subset(ALL.D145, ontology=="CC")
CC.D145 <- CC.D145[order(-CC.D145$numDEInCat),]
BP.D145 <- subset(ALL.D145, ontology=="BP")
BP.D145 <- BP.D145[order(-BP.D145$numDEInCat),]
write.csv(ALL.D145 , file = "RAnalysis/Output/ALL_Sig_Enriched_GO.05_D145.csv")
write.csv(MF.D145 , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_D145.csv")
write.csv(CC.D145 , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_D145.csv")
write.csv(BP.D145 , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_D145.csv")


```


```{r}
##### DMGs in all comparisons #####

ints_10_135 <- merge(D10.sig, D135.sig, by="gene")
ints_10_145 <- merge(D10.sig, D145.sig, by="gene")
ints_135_145 <- merge(D135.sig, D145.sig, by="gene")
ints_all <- merge(ints_10_135, D145.sig, by="gene")
ints_all.annot <- merge(ints_all, Annot, by="gene", all.x=TRUE)
ints_all.annot <- ints_all.annot[!duplicated(ints_all.annot$gene),]
write.table(ints_all.annot, 'RAnalysis/Output/Table_S5_DMGs_consistent_INT.tsv', sep='\t', row.names=FALSE)

```

```{r}
# GO Enrichment Analysis of DMG present in all comparisons
#### GO enrichment of DMGs #####

#ALL= should be all possible genes in D0, D10, D135, and D145

ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(ints_all$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.INT <- merge(ALL, Gene.GO.IDs,by="gene", all=T) #merge GO, ID
GO.INT[is.na(GO.INT)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.INT$GO.IDs), "; ") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.INT$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms,
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.INT <- subset(enriched.GO.05, ontology=="MF")
MF.INT <- MF.INT[order(-MF.INT$numDEInCat),]
CC.INT <- subset(enriched.GO.05, ontology=="CC")
CC.INT <- CC.INT[order(-CC.INT$numDEInCat),]
BP.INT <- subset(enriched.GO.05, ontology=="BP")
BP.INT <- BP.INT[order(-BP.INT$numDEInCat),]
write.csv(MF.INT , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_INT.csv")
write.csv(CC.INT , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_INT.csv")
write.csv(BP.INT , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_INT.csv")
```

```{r}
##### Heatmap of DMG in Ambient by time #####
Time.Amb.meth_table.means <- aggregate(per.meth ~ TimePoint*gene, data=MD.Time, FUN=mean)
Time.Amb.meth_table.means <- Time.Amb.meth_table.means[Time.Amb.meth_table.means$gene %in% Time.Amb.sig$gene,]

Time.Amb <- Time.Amb.meth_table.means %>% pivot_wider(names_from = c(TimePoint), values_from = per.meth)
Time.Amb.mat <- as.matrix(Time.Amb[,-1])

pdf("RAnalysis/Output/Time.DMG_Rel_HEATMAP.pdf")
TimeHM <- heatmap.2(Time.Amb.mat, margins = c(10,20), cexRow = 1.2, cexCol = 1, Colv=FALSE, dendrogram="row", labRow=FALSE, col = rev(RColorBrewer::brewer.pal(name = "RdYlBu", n = 10)), na.color = "black", density.info = "none", trace = "none", scale = "row", sepwidth=c(0.025,0.025), sepcolor="darkgray", keysize = 1)
dev.off()

```

```{r}
##### Heatmap of DMG in D10 #####
D10.meth_table.means <- aggregate(per.meth ~ Initial.Treatment*gene, data=MD.D10, FUN=mean)
D10.meth_table.means <- D10.meth_table.means[D10.meth_table.means$gene %in% D10.sig$gene,]

D10 <- D10.meth_table.means %>% pivot_wider(names_from = c(Initial.Treatment), values_from = per.meth)
D10.mat <- as.matrix(D10[,-1])

col.order <- c( "Ambient",
                "Low", 
                "Super.Low")
D10.mat <- D10.mat[,col.order]

pdf("RAnalysis/Output/D10.DMG_Rel_HEATMAP.pdf")
D10HM <- heatmap.2(D10.mat, margins = c(10,20), cexRow = 1.2, cexCol = 1, Colv=FALSE, dendrogram="row", labRow=FALSE, col = rev(RColorBrewer::brewer.pal(name = "RdYlBu", n = 10)), na.color = "black", density.info = "none", trace = "none", scale = "row", sepwidth=c(0.025,0.025), sepcolor="darkgray", keysize = 1)
dev.off()

```

```{r}
##### Heatmap of DMG in D135 #####
D135.meth_table.means <- aggregate(per.meth ~ Initial.Treatment*gene, data=MD.D135, FUN=mean)
D135.meth_table.means <- D135.meth_table.means[D135.meth_table.means$gene %in% D135.sig$gene,]

D135 <- D135.meth_table.means %>% pivot_wider(names_from = c(Initial.Treatment), values_from = per.meth)
D135.mat <- as.matrix(D135[,-1])

col.order <- c( "Ambient",
                "Low", 
                "Super.Low")
D135.mat <- D135.mat[,col.order]

pdf("RAnalysis/Output/D135.DMG_Rel_HEATMAP.pdf")
D135HM <- heatmap.2(D135.mat, margins = c(10,20), cexRow = 1.2, cexCol = 1, Colv=FALSE, dendrogram="row", labRow=FALSE, col = rev(RColorBrewer::brewer.pal(name = "RdYlBu", n = 10)), na.color = "black", density.info = "none", trace = "none", scale = "row", sepwidth=c(0.025,0.025), sepcolor="darkgray", keysize = 1)
dev.off()
```

```{r}
##### Heatmap of DMG in D145 #####
D145.meth_table.means <- aggregate(per.meth ~ gene*Initial.Treatment*Secondary.Treatment, data=MD.D145, FUN=mean)
D145.meth_table.means <- D145.meth_table.means[D145.meth_table.means$gene %in% D145.sig$gene,]

D145 <- D145.meth_table.means %>% pivot_wider(names_from = c(Initial.Treatment, Secondary.Treatment), values_from = per.meth)
D145.mat <- as.matrix(D145[,-1])

col.order <- c("Ambient_Ambient", "Ambient_Low", "Low_Ambient", "Low_Low", "Super.Low_Ambient", "Super.Low_Low")
D145.mat  <- D145.mat[,col.order]

pdf("RAnalysis/Output/D145.DMG_Rel_HEATMAP.pdf")
D145HM <- heatmap.2(D145.mat, margins = c(10,20), cexRow = 1.2, cexCol = 1, Colv=FALSE, dendrogram="row", labRow=FALSE, col = rev(RColorBrewer::brewer.pal(name = "RdYlBu", n = 10)), na.color = "black", density.info = "none", trace = "none", scale = "row", sepwidth=c(0.1,0.1), sepcolor="white", keysize = 1, colsep=c(0,2,4,6))
dev.off()

```

```{r}
### Magnitude of change in secondary exposure

D145.mat.1 <- D145.mat[,c(1:2)]
D145.mat.1 <- as.data.frame(D145.mat.1)
D145.mat.1$gene <-rownames(D145.mat.1)
D145.mat.1$diff <- D145.mat.1$Ambient_Low - D145.mat.1$Ambient_Ambient
mean(D145.mat.1$diff)
std.error(D145.mat.1$diff)
var(D145.mat.1$diff)

D145.mat.2 <- D145.mat[,c(3:4)]
D145.mat.2 <- as.data.frame(D145.mat.2)
D145.mat.2$gene <-rownames(D145.mat.2)
D145.mat.2$diff <- D145.mat.2$Low_Low - D145.mat.2$Low_Ambient
mean(D145.mat.2$diff)
std.error(D145.mat.2$diff)
var(D145.mat.2$diff)

D145.mat.3 <- D145.mat[,c(5:6)]
D145.mat.3 <- as.data.frame(D145.mat.3)
D145.mat.3$gene <-rownames(D145.mat.3)
D145.mat.3$diff <- D145.mat.3$Super.Low_Low - D145.mat.3$Super.Low_Ambient
mean(D145.mat.3$diff)
std.error(D145.mat.3$diff)
var(D145.mat.3$diff)

labs1 <- c(2.2, -0.4, -0.5)
labs2 <- c(0.4, 0.3, 0.3)

pdf("RAnalysis/Output/D145.DMG_Rel_HEATMAP_withDiff.pdf")
heatmap.2(D145.mat, margins = c(10,20), cexRow = 1.2, cexCol = 1, Colv=FALSE, dendrogram="row", labRow=FALSE, col = rev(RColorBrewer::brewer.pal(name = "RdYlBu", n = 10)), na.color = "black", density.info = "none", trace = "none", scale = "row", sepwidth=c(0.1,0.1), sepcolor="white", keysize = 1, colsep=c(0,2,4,6), add.expr=text(x=c(2,4,6), y=c(420,420,420), labels=labs1))
dev.off()
```


```{r}
##### Gene Ontology Plots
BPplot.10 <- BP.D10 %>% head(., n = 10L) %>%
  #filter(numDEInCat > 3) %>%
  ggplot(aes(x=reorder(term, numDEInCat), y= numDEInCat) ) +
      geom_segment(aes(x=reorder(term, numDEInCat) ,xend=reorder(term, -numDEInCat), y=0, yend=numDEInCat), color="black") +
      geom_point(size=1, color="black") +
  coord_flip()+
      theme(
        panel.grid.minor.y = element_blank(),
        panel.grid.major.y = element_blank(),
        legend.position="none") +
  xlab("") +
  ylab("") +
  ylim(0,20) +
  ggtitle("Day 10 Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
                     panel.grid.major = element_blank(), #Set major gridlines
                     panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) + #Set the plot background
        theme(axis.text = element_text(size=8) )#Set axis text size
BPplot.10

BPplot.135 <- BP.D135 %>% head(., n = 10L) %>%
  #filter(numDEInCat > 3) %>%
  ggplot(aes(x=reorder(term, numDEInCat), y= numDEInCat) ) +
      geom_segment(aes(x=reorder(term, numDEInCat) ,xend=reorder(term, -numDEInCat), y=0, yend=numDEInCat), color="black") +
      geom_point(size=1, color="black") +
  coord_flip()+
      theme(
        panel.grid.minor.y = element_blank(),
        panel.grid.major.y = element_blank(),
        legend.position="none") +
  xlab("") +
  ylab("") +
  ylim(0,20) +
  ggtitle("Day 135 Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
                     panel.grid.major = element_blank(), #Set major gridlines
                     panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) + #Set the plot background
        theme(axis.text = element_text(size=8) )#Set axis text size
BPplot.135

BPplot.145 <- BP.D145 %>% head(., n = 10L) %>%
  #filter(numDEInCat > 3) %>%
  ggplot(aes(x=reorder(term, numDEInCat), y= numDEInCat) ) +
      geom_segment(aes(x=reorder(term, numDEInCat) ,xend=reorder(term, -numDEInCat), y=0, yend=numDEInCat), color="black") +
      geom_point(size=1, color="black") +
  coord_flip()+
      theme(
        panel.grid.minor.y = element_blank(),
        panel.grid.major.y = element_blank(),
        legend.position="none") +
  xlab("") +
  ylab("") +
  ylim(0,20) +
  ggtitle("Day 145 Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
                     panel.grid.major = element_blank(), #Set major gridlines
                     panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) + #Set the plot background
        theme(axis.text = element_text(size=8) )#Set axis text size
BPplot.145


plot_grid(BPplot.10, BPplot.135, BPplot.145, ncol=1, align="v")
ggsave("RAnalysis/Output/BP_GOPlots.pdf")
```


**Genes in top BP GO for each sampling period**
```{r}


BP.D10[1,1]
BP.D135[1,1]
BP.D145[1,1]

D10.topBPGO.genes <- D10.sig.annot[grep(as.character(BP.D10[1,1]),D10.sig.annot$GO.IDs), ]
D135.topBPGO.genes <- D135.sig.annot[grep(as.character(BP.D135[1,1]),D135.sig.annot$GO.IDs), ]
D145.topBPGO.genes <- D145.sig.annot[grep(as.character(BP.D145[1,1]),D145.sig.annot$GO.IDs), ]

#Day 10 
for (i in 1:nrow(D10.topBPGO.genes)){
  sig.gene <- subset(MD.D10, gene==D10.topBPGO.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth
  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    ylim(0,100)+ #set y limits to 0 and 100%
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()
  ggsave(temp_plot, file=paste0("RAnalysis/Output/NegRegGeneExpress_D10plot_", D10.topBPGO.genes$gene[i],"_",D10.topBPGO.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}

#Day 135 
for (i in 1:nrow(D135.topBPGO.genes)){
  sig.gene <- subset(MD.D135, gene==D135.topBPGO.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth
  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    ylim(0,100)+ #set y limits to 0 and 100%
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()
  ggsave(temp_plot, file=paste0("RAnalysis/Output/NegRegTransExpress_D135plot_", D135.topBPGO.genes$gene[i],"_",D135.topBPGO.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}

#Day 145 
for (i in 1:nrow(D145.topBPGO.genes)){
  sig.gene <- subset(MD.D145, gene ==D145.topBPGO.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth
  temp_plot <- ggplot(data=means, aes(x=Secondary.Treatment, y=per.meth, group=Initial.Treatment, colour=Initial.Treatment, shape=Initial.Treatment)) + #plot data
    geom_line(size=0.7, position=position_dodge(.1)) + #plot lines
    scale_colour_manual(values = c("blue", "purple", "red")) + #set line color
    geom_point(size=4, position=position_dodge(.1), colour="black") + #set point characteristics
    scale_shape_manual(values=c(1,18,16)) + #set shapes
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), #plot error bars
                  width=0, position=position_dodge(.1), colour="black") + #set error bar characteristics
    xlab("Secondary Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    theme_bw()
  ggsave(temp_plot, file=paste0("RAnalysis/Output/PosRegTrans_D145plot_", D145.topBPGO.genes$gene[i],"_",D145.topBPGO.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}
```

