---
title: "Geoduck DNA Methylation"
author: "HPutnam"
date: "12/18/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load Libraries

```{r message = FALSE, warning = FALSE}
library(plotrix) 
library(ggplot2)
library(gridExtra)
library(pheatmap)
library(dplyr)
#devtools::install_github("tidyverse/tidyr")
library(tidyverse)
library(tidyr)
library(topGO)
library(goseq)
#library(methylGSA)
library(genefilter)

```

# Load Genomic Features and Sample Info

```{r}
#load sample information
sample.info <- read.csv("/home/hputnam/Geoduck_BS/RAnalysis/Data/Sample.Info.csv", header=T, sep=",", na.string="NA", stringsAsFactors = F) #read in info
samp <- sample.info$Sample.ID # set sample info
samp <- gsub("[_]", "-", samp) #remove extra characters

#load genes gff
Genes <- read.csv("/home/hputnam/Geoduck_BS/RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3", header=F, sep="\t", na.string="NA", skip=3, stringsAsFactors = F) #read in data file
Genes <- Genes[,c(1,4,5,9)] #select desired columns only
colnames(Genes) <- c("scaffold", "start", "stop", "gene") #rename columns
Genes$gene <- gsub(";.*","",Genes$gene) #remove extra characters
Genes$gene <- gsub("ID=","",Genes$gene) #remove extra characters

#Load annotation file
Annot <- read.csv("/home/hputnam/Geoduck_BS/RAnalysis/Data/Genome/Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab", header=TRUE, sep="\t", na.string="NA", stringsAsFactors = F, skip=12)
colnames(Annot) <- c("gene","start", "end", "score", "uniprot.id", "Match.ID", "m_start", "m_end", "E.value", "identity", "al")
Annot$gene <- gsub("21910-","",Annot$gene) #remove extra characters
Annot$gene  <- gsub(".m01","",Annot$gene) #remove extra characters
Uniprots <- Annot[,c(1,5)] #pull out uniprot information
colnames(Uniprots) <- c("gene", "uniprot.id") #rename columns
write.csv(Uniprots, "/home/hputnam/Geoduck_BS/RAnalysis/Data/Genome/Uniprot.IDS.csv") #save to file

#Obtain GO terms from Uniprot using Uniprot.IDS.csv https://www.uniprot.org/uploadlists/

GOs <- read.csv("/home/hputnam/Geoduck_BS/RAnalysis/Data/Genome/Uniprot_GOs.csv", header=TRUE, sep=",", na.string="NA", stringsAsFactors = F) #read in gene ontology information
colnames(GOs) <- c("uniprot.id", "protein.name.short", "protein.name.long", "gene.names", "organism", "GO.IDs") #rename columns of gene ontology information

GO.data <- merge(Annot, GOs, by="uniprot.id")

```

# Filter and Concatenate Datasets

```{bash}
#Time dataset
mkdir /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time 

multiIntersectBed -i \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-41_S38_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-42_S39_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-43_S40_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-44_S41_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-119_S31_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-120_S32_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-135_S35_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-136_S36_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-151_S2_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-152_S3_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-153_S4_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-154_S5_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-181_S16_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-182_S17_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-184_S18_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-185_S19_L003_5x.tab \
> /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/time.5x.bed

cat /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/time.5x.bed | awk '$4 ==16' > /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/time.filtered.16.5x.bed 

cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-41_S38_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-42_S39_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-43_S40_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-44_S41_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-119_S31_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-120_S32_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-135_S35_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-136_S36_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-151_S2_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-152_S3_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-153_S4_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-154_S5_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-181_S16_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-182_S17_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-184_S18_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-185_S19_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file that ends in posOnly
#b: annotated gene list
#Save output in a new file that has the same base name and ends in -Annotated.txt
for f in /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b /home/hputnam/Geoduck_BS/RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/*_gene
do
  intersectBed \
  -a ${f} \
  -b /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/time.filtered.16.5x.bed  \
  > ${f}_CpG_time.bed
done

wc -l /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/*_CpG_time.bed
```

```{bash}
# Day 10 Data Set
mkdir /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10 

multiIntersectBed -i \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-103_S27_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-104_S28_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-111_S29_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-113_S30_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-119_S31_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-120_S32_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-127_S33_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-128_S34_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-135_S35_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-136_S36_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-143_S37_L005_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-145_S38_L005_5x.tab \
> /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/D10.5x.bed

cat /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/D10.5x.bed | awk '$4 ==12' > /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/D10.filtered.12.5x.bed 

cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-103_S27_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-104_S28_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-111_S29_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-113_S30_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-119_S31_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-120_S32_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-127_S33_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-128_S34_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-135_S35_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-136_S36_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-143_S37_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-145_S38_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b /home/hputnam/Geoduck_BS/RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/*_gene
do
  intersectBed \
  -a ${f} \
  -b /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/D10.filtered.12.5x.bed   \
  > ${f}_CpG_D10.bed
done

wc -l /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/*_CpG_D10.bed
```

```{bash}
# Day 135 Data

mkdir /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135 

multiIntersectBed -i \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-151_S2_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-152_S3_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-153_S4_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-154_S5_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-159_S6_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-160_S7_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-161_S8_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-162_S9_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-167_S10_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-168_S11_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-169_S12_L002_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-170_S13_L002_5x.tab \
> /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/D135.5x.bed

cat /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/D135.5x.bed | awk '$4 ==12' > /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/D135.filtered.12.5x.bed 

cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-103_S27_L005_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-151_S2_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-152_S3_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-153_S4_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-154_S5_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-159_S6_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-160_S7_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-161_S8_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-162_S9_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-167_S10_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-168_S11_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-169_S12_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-170_S13_L002_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b /home/hputnam/Geoduck_BS/RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/*_gene
do
  intersectBed \
  -a ${f} \
  -b /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/D135.filtered.12.5x.bed   \
  > ${f}_CpG_D135.bed
done

wc -l /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/*_CpG_D135.bed
```

```{bash}
# Day 145 Data

mkdir /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145 

multiIntersectBed -i \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-175_S14_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-176_S15_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-181_S16_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-182_S17_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-184_S18_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-185_S19_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-187_S20_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-188_S21_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-193_S22_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-194_S23_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-199_S24_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-200_S25_L003_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-205_S26_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-206_S27_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-208_S28_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-209_S29_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-214_S30_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-215_S31_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-220_S32_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-221_S33_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-226_S34_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-227_S35_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-229_S36_L004_5x.tab \
/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-230_S37_L004_5x.tab \
> /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/D145.5x.bed

cat /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/D145.5x.bed | awk '$4 ==24' > /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/D145.filtered.24.5x.bed 

cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-175_S14_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-176_S15_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-181_S16_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-182_S17_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-184_S18_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-185_S19_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-187_S20_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-188_S21_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-193_S22_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-194_S23_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-199_S24_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-200_S25_L003_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-205_S26_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-206_S27_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-208_S28_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-209_S29_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-214_S30_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-215_S31_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-220_S32_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-221_S33_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-226_S34_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-227_S35_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-229_S36_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/
cp /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/EPI-230_S37_L004_5x.tab /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b /home/hputnam/Geoduck_BS/RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/*_gene
do
  intersectBed \
  -a ${f} \
  -b /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/D145.filtered.24.5x.bed   \
  > ${f}_CpG_D145.bed
done

wc -l /home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/*_CpG_D145.bed
```

# Load in methylation counts

```{r}
# Time
meth.data.T <- list.files(path = "/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time", pattern = "time.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=F, sep="\t", na.string="NA", stringsAsFactors = F) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.T) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.T$gene <- gsub(";.*","",meth.data.T$gene) #remove extra characters
meth.data.T$gene <- gsub("ID=","",meth.data.T$gene) #remove extra characters
meth.data.T$Sample.ID <- gsub("/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/Time/","",meth.data.T$Sample.ID) #remove extra characters
meth.data.T$Sample.ID <- gsub("_.*","",meth.data.T$Sample.ID) #remove extra characters 
meth.data.T$Sample.ID <- gsub("-","_",meth.data.T$Sample.ID) #remove extra characters

MD.Time <- merge(meth.data.T,sample.info, by="Sample.ID")
```

```{r}
#Day 10
meth.data.D10 <- list.files(path = "/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10", pattern = "D10.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=F, sep="\t", na.string="NA", stringsAsFactors = F) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.D10) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D10$gene <- gsub(";.*","",meth.data.D10$gene) #remove extra characters
meth.data.D10$gene <- gsub("ID=","",meth.data.D10$gene) #remove extra characters
meth.data.D10$Sample.ID <- gsub("/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D10/","",meth.data.D10$Sample.ID) #remove extra characters
meth.data.D10$Sample.ID <- gsub("_.*","",meth.data.D10$Sample.ID) #remove extra characters 
meth.data.D10$Sample.ID <- gsub("-","_",meth.data.D10$Sample.ID) #remove extra characters

MD.D10 <- merge(meth.data.D10, sample.info, by="Sample.ID")
```

```{r}
#Day 135
meth.data.D135 <- list.files(path = "/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135", pattern = "D135.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=F, sep="\t", na.string="NA", stringsAsFactors = F) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.D135) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D135$gene <- gsub(";.*","",meth.data.D135$gene) #remove extra characters
meth.data.D135$gene <- gsub("ID=","",meth.data.D135$gene) #remove extra characters
meth.data.D135$Sample.ID <- gsub("/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D135/","",meth.data.D135$Sample.ID) #remove extra characters
meth.data.D135$Sample.ID <- gsub("_.*","",meth.data.D135$Sample.ID) #remove extra characters 
meth.data.D135$Sample.ID <- gsub("-","_",meth.data.D135$Sample.ID) #remove extra characters

MD.D135 <- merge(meth.data.D135, sample.info, by="Sample.ID")
```

```{r}
#Day 145
meth.data.D145 <- list.files(path = "/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145", pattern = "D145.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=F, sep="\t", na.string="NA", stringsAsFactors = F) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.D145) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D145$gene <- gsub(";.*","", meth.data.D145$gene) #remove extra characters
meth.data.D145$gene <- gsub("ID=","",meth.data.D145$gene) #remove extra characters
meth.data.D145$Sample.ID <- gsub("/home/hputnam/Geoduck_BS/RAnalysis/Data/OSF/D145/","",meth.data.D145$Sample.ID) #remove extra characters
meth.data.D145$Sample.ID <- gsub("_.*","",meth.data.D145$Sample.ID) #remove extra characters 
meth.data.D145$Sample.ID <- gsub("-","_",meth.data.D145$Sample.ID) #remove extra characters

MD.D145 <- merge(meth.data.D145, sample.info, by="Sample.ID")
```

# Testing for Differentially Methylated Genes

```{r}
# Comparison of Ambient samples through Time
# Binomial GLM to test for differentially methylated genes

sub_meth_table  <- MD.Time
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to stored results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ TimePoint, 
             data=sub_meth_table1, family=binomial)
  a <- anova(fit, test="Chisq")
  
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3], #uncomment if you want to include position of CpG within a gene
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4], #uncomment if you want to include position of CpG within a gene interaction with treatment
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}

results[is.na(results)] <- 0

results$adj.pval.TimePoint <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH') #uncomment if you want to include position of CpG within a gene
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH') #uncomment if you want to include position of CpG within a gene interaction with treatment

Time.Amb.sig <-results
Time.Amb.sig <- Time.Amb.sig[,c(1,3)]
Time.Amb.sig <- Time.Amb.sig[order(Time.Amb.sig$adj.pval.TimePoint),]
Time.Amb.sig <- Time.Amb.sig[which(Time.Amb.sig$adj.pval.TimePoint<0.05), ]

write.table(Time.Amb.sig, '/home/hputnam/Geoduck_BS/RAnalysis/Output/Time.Amb_HP_GLM_SortedResults.tsv', sep='\t', row.names=FALSE)

sub_meth_table$per.meth <- (sub_meth_table$meth/(sub_meth_table$meth+sub_meth_table$unmeth))*100 #calculate percent methylation

# Annotation of DMG under ambient conditions through time
Time.Amb.annot <- merge(Time.Amb.sig , Annot, by="gene", all.x=TRUE)
Time.Amb.annot <- merge(Time.Amb.annot, GOs, by="uniprot.id", all.x=TRUE)
Time.Amb.annot <- Time.Amb.annot[order(Time.Amb.annot$adj.pval.TimePoint),]
Time.Amb.annot <- Time.Amb.annot[!duplicated(Time.Amb.annot$gene),]
write.table(Time.Amb.annot, '/home/hputnam/Geoduck_BS/RAnalysis/Output/Table_S1_Time_Amb_DMG_annot.tsv', sep='\t', row.names=FALSE)
```
```{r}
# GO Enrichment Analysis of Ambient samples through Time
##### GO enrichment of DMGs #####
ALL<-as.data.frame(unique(MD.Time$gene)) #set the all transcripts list
colnames(ALL) <-c("gene")
DMG <- as.character(Time.Amb.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.Time <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.Time[is.na(GO.Time)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.Time$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.Time$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.TIME <- subset(enriched.GO.05, ontology=="MF")
MF.TIME <- MF.TIME[order(-MF.TIME$numDEInCat),]
CC.TIME <- subset(enriched.GO.05, ontology=="CC")
CC.TIME <- CC.TIME[order(-CC.TIME$numDEInCat),]
BP.TIME <- subset(enriched.GO.05, ontology=="BP")
BP.TIME <- BP.TIME[order(-BP.TIME$numDEInCat),]
write.csv(MF.TIME , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/MF_Sig_Enriched_GO.05_TIME.csv")
write.csv(CC.TIME , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/CC_Sig_Enriched_GO.05_TIME.csv")
write.csv(BP.TIME , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/BP_Sig_Enriched_GO.05_TIME.csv")
```


```{r}
# Comparison of Treatments at Day 10
# Binomial GLM to test for differentially methylated genes

meth_table  <- MD.D10
sub_meth_table <- meth_table[meth_table$Initial.Treatment == 'Ambient' | meth_table$Initial.Treatment == 'Low' | meth_table$Initial.Treatment == 'Super.Low', ]
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)
  
#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to stored results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment, 
             data=sub_meth_table1, family=binomial, maxit = 100)
  a <- anova(fit, test="Chisq")
  
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3], #uncomment if you want to include position of CpG within a gene
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4], #uncomment if you want to include position of CpG within a gene interaction with treatment
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}

results[is.na(results)] <- 0

results$adj.pval.Initial.Treatment <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH') #uncomment if you want to include position of CpG within a gene
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH') #uncomment if you want to include position of CpG within a gene interaction with treatment

D10.sig <-results
D10.sig <- D10.sig[,c(1,3)]
D10.sig <- D10.sig[order(D10.sig$adj.pval.Initial.Treatment),]
D10.sig <- D10.sig[which(D10.sig$adj.pval.Initial.Treatment<0.05), ]

write.table(D10.sig, '/home/hputnam/Geoduck_BS/RAnalysis/Output/D10_GLM_SortedResults.tsv', sep='\t', row.names=FALSE)

# Annotation of D10 DMG 
D10.sig.annot <- merge(D10.sig, Annot, by="gene", all.x=TRUE)
D10.sig.annot <- merge(D10.sig.annot, GOs, by="uniprot.id", all.x=TRUE)
D10.sig.annot <- D10.sig.annot[order(D10.sig.annot$adj.pval.Initial.Treatment),]
D10.sig.annot <- D10.sig.annot[!duplicated(D10.sig.annot$gene),]
write.table(D10.sig.annot, '/home/hputnam/Geoduck_BS/RAnalysis/Output/Table_S2_D10_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 10
##### GO enrichment of DMGs #####
ALL<-as.data.frame(unique(MD.D10$gene)) #set the all transcripts list
colnames(ALL) <-c("gene")
DMG <- as.character(D10.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D10 <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.D10[is.na(GO.D10)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D10$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D10$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.D10 <- subset(enriched.GO.05, ontology=="MF")
MF.D10 <- MF.D10[order(-MF.D10$numDEInCat),]
CC.D10 <- subset(enriched.GO.05, ontology=="CC")
CC.D10 <- CC.D10[order(-CC.D10$numDEInCat),]
BP.D10 <- subset(enriched.GO.05, ontology=="BP")
BP.D10 <- BP.D10[order(-BP.D10$numDEInCat),]
write.csv(MF.D10 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/MF_Sig_Enriched_GO.05_D10.csv")
write.csv(CC.D10 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/CC_Sig_Enriched_GO.05_D10.csv")
write.csv(BP.D10 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/BP_Sig_Enriched_GO.05_D10.csv")
```

```{r}
# Comparison of Treatments at Day 135
# Binomial GLM to test for differentially methylated genes

meth_table  <- MD.D135
sub_meth_table <- meth_table[meth_table$Initial.Treatment == 'Ambient' | meth_table$Initial.Treatment == 'Low' | meth_table$Initial.Treatment == 'Super.Low', ]

sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to store results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  #if(length(unique(sub_meth_table1$position)) >1){
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment, 
             data=sub_meth_table1, family=binomial)
  #} else {
  #  # fit glm model
  #  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment * position, 
  #             data=sub_meth_table1, family=binomial)
  #}
  #s <- step(fit, trace=0)
  a <- anova(fit, test="Chisq")
  a
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3],
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4],
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}

results[is.na(results)] <- 0

results$adj.pval.treatment <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH')
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH')

D135.sig <-results
D135.sig <- D135.sig[,c(1,3)]
D135.sig <- D135.sig[order(D135.sig$adj.pval.treatment),]
D135.sig <- D135.sig[which(D135.sig$adj.pval.treatment<0.05), ]
write.table(D135.sig, '/home/hputnam/Geoduck_BS/RAnalysis/Output/D135_GLM_BH_0_05.tsv', sep='\t', row.names=FALSE)

# Annotation of D135 DMG 
D135.sig.annot <- merge(D135.sig, Annot, by="gene", all.x=TRUE)
D135.sig.annot <- merge(D135.sig.annot, GOs, by="uniprot.id", all.x=TRUE)
D135.sig.annot <- D135.sig.annot[order(D135.sig.annot$adj.pval.treatment),]
D135.sig.annot <- D135.sig.annot[!duplicated(D135.sig.annot$gene),]
write.table(D135.sig.annot, '/home/hputnam/Geoduck_BS/RAnalysis/Output/Table_S3_D135_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 135
##### GO enrichment of DMGs #####
ALL<-as.data.frame(unique(MD.D135$gene)) #set the all transcripts list
colnames(ALL) <-c("gene")
DMG <- as.character(D135.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D135 <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.D135[is.na(GO.D135)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D135$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D135$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.D135 <- subset(enriched.GO.05, ontology=="MF")
MF.D135 <- MF.D135[order(-MF.D135$numDEInCat),]
CC.D135 <- subset(enriched.GO.05, ontology=="CC")
CC.D135 <- CC.D135[order(-CC.D135$numDEInCat),]
BP.D135 <- subset(enriched.GO.05, ontology=="BP")
BP.D135 <- BP.D135[order(-BP.D135$numDEInCat),]
write.csv(MF.D135 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/MF_Sig_Enriched_GO.05_D135.csv")
write.csv(CC.D135 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/CC_Sig_Enriched_GO.05_D135.csv")
write.csv(BP.D135 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/BP_Sig_Enriched_GO.05_D135.csv")
```

```{r}
# Comparison of Treatments at Day 145
# Binomial GLM to test for differentially methylated genes
meth_table <- MD.D145
sub_meth_table <- meth_table[meth_table$Initial.Treatment == "Ambient" | meth_table$Initial.Treatment == "Low" | meth_table$Initial.Treatment == "Super.Low"
                             | meth_table$Secondary.Treatment == "Ambient" | meth_table$Secondary.Treatment == "Low", ]
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to store results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
    
    #subset the dataframe gene by gene
    sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
    
    # fit glm position model
    fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment * Secondary.Treatment, 
               data=sub_meth_table1, family=binomial)

    a <- anova(fit, test="LRT")
    
    # capture summary stats to data frame
    df <- data.frame(gene = sub_meth_table1[1,7],
                     pval.treatment1 = a$`Pr(>Chi)`[2],
                     pval.treatment2 = a$`Pr(>Chi)`[3],
                     #position = a$`Pr(>Chi)`[2], 
                     pval.treatment1_x_treatment2 = a$`Pr(>Chi)`[4],
                     #pval.treatment1_x_position = a$`Pr(>Chi)`[2],
                     #pval.treatment2_x_position = a$`Pr(>Chi)`[2],
                     #pval.treatment1_x_pval.treatment2_x_position = a$`Pr(>Chi)`[2],
                     stringsAsFactors = F)

    # bind rows of temporary data frame to the results data frame
    results <- rbind(results, df)
    
  }

results[is.na(results)] <- 0

results$adj.pval.treatment1 <- p.adjust(results$pval.treatment1, method='BH')
results$adj.pval.treatment2 <- p.adjust(results$pval.treatment2, method='BH')
results$adj.pval.treatment1_x_treatment2 <- p.adjust(results$pval.treatment1_x_treatment2, method='BH')
#results$adj.pval.treatment1_x_position <- p.adjust(results$pval.treatment1_x_position, method='BH')
#results$adj.pval.treatment2_x_position <- p.adjust(results$pval.treatment2_x_position, method='BH')
#results$adj.pval.treatment1_x_pval.treatment2_x_position <- p.adjust(results$pval.treatment1_x_pval.treatment2_x_position, method='BH')

D145.sig <-results
D145.sig <- D145.sig[,c(1,5:7)]
D145.sig <- D145.sig[order(results$adj.pval.treatment1_x_treatment2),]
D145.sig <- D145.sig[which(D145.sig$adj.pval.treatment1_x_treatment2<0.05), ]
write.table(D145.sig, '/home/hputnam/Geoduck_BS/RAnalysis/Output/Interaction_D145_GLM_BH_0_05.tsv', sep='\t', row.names=FALSE)

sub_meth_table$per.meth <- (sub_meth_table$meth/(sub_meth_table$meth+sub_meth_table$unmeth))*100
sub_meth_table.D145 <- sub_meth_table

# Annotation of Secondary Exposure Interaction DMG 
D145.sig.annot <- merge(D145.sig, Annot, by="gene", all.x=TRUE)
D145.sig.annot <- merge(D145.sig.annot, GOs, by="uniprot.id", all.x=TRUE)
D145.sig.annot <- D145.sig.annot[order(D145.sig.annot$adj.pval.treatment1_x_treatment2),]
D145.sig.annot <- D145.sig.annot[!duplicated(D145.sig.annot$gene),]
write.table(D145.sig.annot, '/home/hputnam/Geoduck_BS/RAnalysis/Output/Table_S4_D145_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 145
##### GO enrichment of DMGs #####
ALL<-as.data.frame(unique(MD.D145$gene)) #set the all transcripts list
colnames(ALL) <-c("gene")
DMG <- as.character(D145.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D145 <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.D145[is.na(GO.D145)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D145$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D145$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.D145 <- subset(enriched.GO.05, ontology=="MF")
MF.D145 <- MF.D145[order(-MF.D145$numDEInCat),]
CC.D145 <- subset(enriched.GO.05, ontology=="CC")
CC.D145 <- CC.D145[order(-CC.D145$numDEInCat),]
BP.D145 <- subset(enriched.GO.05, ontology=="BP")
BP.D145 <- BP.D145[order(-BP.D145$numDEInCat),]
write.csv(MF.D145 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/MF_Sig_Enriched_GO.05_D145.csv")
write.csv(CC.D145 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/CC_Sig_Enriched_GO.05_D145.csv")
write.csv(BP.D145 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/BP_Sig_Enriched_GO.05_D145.csv")
```


```{r}
##### DMGs in all comparisons #####

ints_10_135 <- merge(D10.sig, D135.sig, by="gene")
ints_10_145 <- merge(D10.sig, D145.sig, by="gene")
ints_135_145 <- merge(D135.sig, D145.sig, by="gene")
ints_all <- merge(ints_10_135, D145.sig, by="gene")
ints_all.annot <- merge(ints_all, Annot, by="gene", all.x=TRUE)
ints_all.annot <- ints_all.annot[!duplicated(ints_all.annot$gene),]
write.table(ints_all.annot, '/home/hputnam/Geoduck_BS/RAnalysis/Output/DMGs_consistent_all_timepoints_annotated.tsv', sep='\t', row.names=FALSE)

```

```{r}
# GO Enrichment Analysis of DMG in all comparisons
##### GO enrichment of DMGs #####

#ALL= all possible genes in D10, D135, and D145
ALL.INT <- unique(rbind(MD.D10$gene,MD.D135$gene,MD.D145$gene))
ALL<-as.data.frame(unique(ints_all$gene)) #set the all transcripts list
colnames(ALL) <-c("gene")
DMG <- as.character(ints_all$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D145 <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.D145[is.na(GO.D145)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D145$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D145$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.D145 <- subset(enriched.GO.05, ontology=="MF")
MF.D145 <- MF.D145[order(-MF.D145$numDEInCat),]
CC.D145 <- subset(enriched.GO.05, ontology=="CC")
CC.D145 <- CC.D145[order(-CC.D145$numDEInCat),]
BP.D145 <- subset(enriched.GO.05, ontology=="BP")
BP.D145 <- BP.D145[order(-BP.D145$numDEInCat),]
write.csv(MF.D145 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/MF_Sig_Enriched_GO.05_D145.csv")
write.csv(CC.D145 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/CC_Sig_Enriched_GO.05_D145.csv")
write.csv(BP.D145 , file = "/home/hputnam/Geoduck_BS/RAnalysis/Output/BP_Sig_Enriched_GO.05_D145.csv")
```

```{r}
##### Heatmap of DMG in Ambient by time #####
Time.Amb.meth_table.means <- aggregate(per.meth ~ TimePoint*gene, data=MD.Time, FUN=mean)
Time.Amb.meth_table.means <- Time.Amb.meth_table.means[Time.Amb.meth_table.means$gene %in% Time.Amb.sig$gene,]

#Time.Amb <- Time.Amb.meth_table.means %>% pivot_wider(names_from = TimePoint, values_from = per.meth)
MD.Time.1 <- MD.Time[,c(1,4,7,12)]

Time.Amb <- reshape(MD.Time.1, 
  timevar = "Sample.ID",
  idvar = c("gene"),
  direction = "wide")

Time.Amb.mat <- Time.Amb[,c(26,28,30,32,2,4,6,8,10,12,14,16,18,20,22,24)]
#Time.Amb.mat <- Time.Amb.mat-rowMeans(Time.Amb.mat)

pheatmap(Time.Amb.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =TRUE, cluster_cols=FALSE,
         show_colnames =TRUE) #plot heatmap of all DEG by group
dev.off()

pdf("/home/hputnam/Geoduck_BS/RAnalysis/Output/Time.Amb.DMG_relative_Meth_All.pdf")
pheatmap(Time.Amb.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =FALSE, cluster_cols=FALSE,
         show_colnames =TRUE) #plot heatmap of all DEG by group
dev.off()
```

```{r}
D10.meth_table.means <- aggregate(per.meth ~ Initial.Treatment*gene, data=MD.D10, FUN=mean)
D10.meth_table.means <- D10.meth_table.means[D10.meth_table.means$gene %in% ints_all$gene,]

#D10 <- D10.meth_table.means %>% pivot_wider(names_from = Initial.Treatment, values_from = per.meth)
#D10 <- D10[D10$gene %in% ints_all.annot$gene,]

D10 <- reshape(D10.meth_table.means, 
  timevar = "Initial.Treatment",
  idvar = c("gene"),
  direction = "wide")

row.names(D10) <- D10$gene

D10.mat <- D10[,2:4]
D10.mat <- D10.mat[order(D10.mat$per.meth.Ambient),]
#D10.mat <- D10.mat-rowMeans(D10.mat[1])

Breaks <- seq(min(0), max(100), length = 100)

pdf("/home/hputnam/Geoduck_BS/RAnalysis/Output/D10.DMG_Rel_Amb_Meth.pdf", width = 1.5, height= 4)
pheatmap(D10.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =TRUE, cluster_cols=FALSE,cluster_rows = FALSE,fontsize_row = 2,
         show_colnames =TRUE, breaks = Breaks) #plot heatmap of all DEG by group
dev.off()
```

```{r}
D135.meth_table.means <- aggregate(per.meth ~ Initial.Treatment*gene, data=MD.D135, FUN=mean)
D135.meth_table.means <- D135.meth_table.means[D135.meth_table.means$gene %in% ints_all$gene,]

D135 <- reshape(D135.meth_table.means, 
  timevar = "Initial.Treatment",
  idvar = c("gene"),
  direction = "wide")

row.names(D135) <- D135$gene

D135.mat <- D135[,2:4]
D135.mat <- D135.mat[match(rownames(D10.mat), rownames(D135.mat)),]
#D135.mat <- D135.mat-rowMeans(D135.mat[1])

pdf("/home/hputnam/Geoduck_BS/RAnalysis/Output/D135.DMG_Rel_Amb_Meth.pdf", width = 1.5, height= 4)
pheatmap(D135.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =TRUE, cluster_cols=FALSE,cluster_rows = FALSE,fontsize_row = 2,
         show_colnames =TRUE, breaks = Breaks) #plot heatmap of all DEG by group
dev.off()
```

```{r}
D145.meth_table.means <- aggregate(per.meth ~ gene*Initial.Treatment*Secondary.Treatment, data=MD.D145, FUN=mean)
D145.meth_table.means <- D145.meth_table.means[D145.meth_table.means$gene %in% ints_all$gene,]
D145.meth_table.means$group <- paste0(D145.meth_table.means$Initial.Treatment, D145.meth_table.means$Secondary.Treatment)
D145.meth_table.means <- D145.meth_table.means[,c(1,4,5)]
#D145 <- D145.meth_table.means %>% pivot_wider(names_from = c(treatment1,treatment2), values_from = per.meth)
D145 <- reshape(D145.meth_table.means, 
  timevar = "group",  
  idvar = c("gene"),
  direction = "wide")

row.names(D145) <- D145$gene
D145.mat <- D145[,2:7]

# D145.mat$per.meth.AmbientAmbient <- D145.mat$per.meth.AmbientAmbient  - D145.mat$per.meth.AmbientAmbient
# D145.mat$per.meth.AmbientLow <- D145.mat$per.meth.AmbientLow  - D145.mat$per.meth.AmbientAmbient
# D145.mat$per.meth.LowAmbient <- D145.mat$per.meth.LowAmbient  - D145.mat$per.meth.LowAmbient
# D145.mat$per.meth.LowLow <- D145.mat$per.meth.LowLow  - D145.mat$per.meth.LowAmbient
# D145.mat$per.meth.Super.LowAmbient <- D145.mat$per.meth.Super.LowAmbient  - D145.mat$per.meth.Super.LowAmbient
# D145.mat$per.meth.Super.LowLow <- D145.mat$per.meth.Super.LowLow  - D145.mat$per.meth.Super.LowAmbient
col.order <- c("per.meth.AmbientAmbient", "per.meth.AmbientLow", "per.meth.LowAmbient", "per.meth.LowLow", "per.meth.Super.LowAmbient", "per.meth.Super.LowLow")
D145.mat  <- D145.mat[,col.order]
D145.mat <- D145.mat[match(rownames(D10.mat), rownames(D145.mat)),]

D145.mat.1 <- D145.mat[,c(1:2)]
D145.mat.2 <- D145.mat[,c(3:4)]
D145.mat.3 <- D145.mat[,c(5:6)]

pdf("/home/hputnam/Geoduck_BS/RAnalysis/Output/D145.DMG_absolute_Meth.pdf", width = 3, height= 5)
pheatmap(D145.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =TRUE, cluster_rows=FALSE, cluster_cols=FALSE, gaps_col = c(2,4),
         fontsize_row = 2, show_colnames =TRUE, breaks = Breaks) #plot heatmap of all DEG by group
dev.off()


# pdf("/home/hputnam/Geoduck_BS/RAnalysis/Output/D145.DMG_Rel_Amb_Meth_1.pdf")
# pheatmap(D145.mat.1, clustering_method = "average", 
#          clustering_distance_rows="euclidean", show_rownames =TRUE, fontsize_row = 2, cluster_cols=FALSE,
#          cluster_rows=F, show_colnames =TRUE) #plot heatmap of all DEG by group
# dev.off()
# pdf("/home/hputnam/Geoduck_BS/RAnalysis/Output/D145.DMG_Rel_Amb_Meth_2.pdf")
# pheatmap(D145.mat.2, clustering_method = "average", 
#          clustering_distance_rows="euclidean", show_rownames =TRUE, fontsize_row = 2, cluster_cols=FALSE,
#          cluster_rows=F, show_colnames =TRUE) #plot heatmap of all DEG by group
# dev.off()
# pdf("/home/hputnam/Geoduck_BS/RAnalysis/Output/D145.DMG_Rel_Amb_Meth_3.pdf")
# pheatmap(D145.mat.3, clustering_method = "average", 
#          clustering_distance_rows="euclidean", show_rownames =TRUE, fontsize_row = 2, cluster_cols=FALSE,
#          cluster_rows=F, show_colnames =TRUE) #plot heatmap of all DEG by group
# dev.off()
```


```{r}

#   means <- aggregate(per.meth ~ Initial.Treatment*gene, data=D10.meth_table.means, FUN=mean)
#   ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
#   means$ses <- ses$per.meth
#   

ints_all.annot <- na.omit(ints_all.annot)

# D10.int <- MD.D10[MD.D10$gene %in% ints_all.annot$gene,]
# D135.int <- MD.D135[MD.D135$gene %in% ints_all.annot$gene,]
# D145.int <- MD.D145[MD.D145$gene %in% ints_all.annot$gene,]
# 
# D10.int.mean <- aggregate(per.meth ~ Initial.Treatment*gene, data=D10.int, FUN=mean)
# D135.int.mean <- aggregate(per.meth ~ Initial.Treatment*gene, data=D135.int, FUN=mean)
# D145.int.mean <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment*gene, data=D145.int, FUN=mean)

for (i in 1:nrow(ints_all.annot)){
  sig.gene <- subset(MD.D10, gene ==ints_all.annot$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()

  ggsave(temp_plot, file=paste0("/home/hputnam/Geoduck_BS/RAnalysis/Output/D10plot_", ints_all.annot$gene[i],"_",ints_all.annot$protein.name[i],".png"), width = 14, height = 10, units = "cm")
}

for (i in 1:nrow(ints_all.annot)){
  sig.gene <- subset(MD.D135, gene ==ints_all.annot$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()

  ggsave(temp_plot, file=paste0("/home/hputnam/Geoduck_BS/RAnalysis/Output/D135plot_", ints_all.annot$gene[i],"_",ints_all.annot$protein.name[i],".png"), width = 14, height = 10, units = "cm")
}

for (i in 1:nrow(ints_all.annot)){
  sig.gene <- subset(sub_meth_table.D145, gene ==ints_all.annot$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(data=means, aes(x=Secondary.Treatment, y=per.meth, group=Initial.Treatment, colour=Initial.Treatment, shape=Initial.Treatment)) + #plot data
    geom_line(size=0.7, position=position_dodge(.1)) + #plot lines
    scale_colour_manual(values = c("blue", "purple", "red")) + #set line color
    geom_point(size=4, position=position_dodge(.1), colour="black") + #set point characteristics
    scale_shape_manual(values=c(1,18,16)) + #set shapes
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), #plot error bars
                  width=0, position=position_dodge(.1), colour="black") + #set error bar characteristics
    xlab("Secondary Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    theme_bw()

  ggsave(temp_plot, file=paste0("/home/hputnam/Geoduck_BS/RAnalysis/Output/D145plot_",ints_all.annot$gene[i],"_", ints_all.annot$protein.name[i],".png"), width = 14, height = 10, units = "cm")
 }
```

```{r}

x <- merge(D10, D135, by="gene")
x <- merge(x, D145, by="gene")

```








