---
title: "Geoduck DNA Methylation"
author: "HPutnam"
date: "12/18/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Load Libraries

```{r message = FALSE, warning = FALSE}
library(plotrix) 
library(ggplot2)
library(gridExtra)
library(seacarb) 
library(pheatmap)
library(dplyr)
library(tidyverse)
library(tidyr)
library(topGO)
library(goseq)
library(genefilter)

```

```{r}
# FLOW RATES
#Initial Exposure: Quantifying flow and testing for differences among treatments
Flow1 <- read.csv("RAnalysis/Data/Flow1_Juvenile_Geoduck.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
#flow values measured in ml/10 sec
Flow1$Rate <- (((Flow1$Rate1 + Flow1$Rate2)/2)*6)/1000*60 #calculate flow rate in liters per hour
Flow1.Rate <- aggregate(Rate ~ Treatment, data=Flow1, FUN=mean) #mean flow by treatment
Flow1.Rate$se <- aggregate(Rate ~ Treatment, data=Flow1, FUN=std.error) #sem flow by treatment
mean(Flow1$Rate) #mean flow across all tanks
std.error(Flow1$Rate) #sem flow across all tanks


FL1 <- aov(Rate ~ Treatment, data=Flow1) #test for differences in flow rate between treatments
FL1.res <- anova(FL1) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(FL1$residuals) #plot histogram of residuals
boxplot(FL1$residuals) #plot boxplot of residuals
plot(FL1) #display residuals versus fitter, normal QQ plot, leverage plot


#Secondary Exposure: Quantifying flow and testing for differences among treatments
Flow2 <- read.csv("RAnalysis/Data/Flow2_Juvenile_Geoduck.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
#flow values measured in ml/15 sec
Flow2$Rate <- ((Flow2$Rate1)*4)/1000*60 #calculate flow rate in liters per hour
Flow2.Rate <- aggregate(Rate ~ Treatment, data=Flow2, FUN=mean) #mean flow by treatment
Flow2.Rate$se <- aggregate(Rate ~ Treatment, data=Flow2, FUN=std.error) #sem flow by treatment
mean(Flow2$Rate) #mean flow across all tanks
std.error(Flow2$Rate) #sem flow across all tanks

FL2 <- aov(Rate ~ Treatment, data=Flow2) #test for differences in flow rate between treatments
FL2.res <-anova(FL2) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(FL2$residuals) #plot histogram of residuals
boxplot(FL2$residuals) #plot boxplot of residuals
plot(FL2) #display residuals versus fitter, normal QQ plot, leverage plot
```

```{r}
# FEEDING RATES
#Initial Exposure: Quantifying feeding and testing for differences among treatments
cells.1 <- read.csv("RAnalysis/Data/Cell_Counts_Juvenile_Geoduck_Exp1.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
cells.1$Avg.Cells <- rowMeans(cells.1[,c("Count1",  "Count2")], na.rm = TRUE) #calculate average of counts
cells.1$cell.num <- cells.1$Avg.Cells/cells.1$Volume.Counted #calculate density

avg.cells.tank.1 <- do.call(data.frame,aggregate(cell.num ~ Tank, data = cells.1, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Tank
avg.cells.trt.1 <- do.call(data.frame,aggregate(cell.num ~ Treatment, data = cells.1, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Treatment
avg.cells.1 <- do.call(data.frame,aggregate(cell.num ~ Volume.Counted, data = cells.1, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Treatment
colnames(avg.cells.tank.1) <- c("Tank", "mean", "se") #rename columns 
colnames(avg.cells.trt.1) <- c("Treatment", "mean", "se") #rename columns 

cells1.tank <- aov(cell.num ~Tank, data=cells.1) #test the hypothesis feeding does not differ between tanks
cells1.tank.res <-anova(cells1.tank) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(cells1.tank$residuals) #plot histogram of residuals
boxplot(cells1.tank$residuals) #plot boxplot of residuals
plot(cells1.tank) #display residuals versus fitter, normal QQ plot, leverage plot

cells1.trt <- aov(cell.num ~Treatment, data=cells.1) #test the hypothesis feeding does not differ between treatments
cells1.trt.res <-anova(cells1.trt) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(cells1.trt$residuals) #plot histogram of residuals
boxplot(cells1.trt$residuals) #plot boxplot of residuals
plot(cells1.trt) #display residuals versus fitter, normal QQ plot, leverage plot

#Secondary Exposure: Quantifying feeding and testing for differences among treatments
cells.2 <- read.csv("RAnalysis/Data/Cell_Counts_Juvenile_Geoduck_Exp2.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
cells.2$cell.num <- rowMeans(cells.2[,c("Count1",  "Count2", "Count3")], na.rm = TRUE) #calculate average of counts
avg.cells.tank.2 <- do.call(data.frame,aggregate(cell.num ~ Tank, data = cells.2, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Tank
avg.cells.trt.2 <- do.call(data.frame,aggregate(cell.num ~ Treatment, data = cells.2, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Treatment
avg.cells.2 <- do.call(data.frame,aggregate(cell.num ~ Notes, data = cells.2, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each Treatment
colnames(avg.cells.tank.2) <- c("Tank", "mean", "se") #rename columns 
colnames(avg.cells.trt.2) <- c("Treatment", "mean", "se") #rename columns 

cells2.tank <- aov(cell.num ~Tank, data=cells.2) #test the hypothesis feeding does not differ between tanks
cells2.tank.res <-anova(cells2.tank) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(cells2.tank$residuals) #plot histogram of residuals
boxplot(cells2.tank$residuals) #plot boxplot of residuals
plot(cells2.tank) #display residuals versus fitter, normal QQ plot, leverage plot

cells2.trt <- aov(cell.num ~Treatment, data=cells.2) #test the hypothesis feeding does not differ between treatments
cells2.trt.res <-anova(cells2.trt) #display anova results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(cells2.trt$residuals) #plot histogram of residuals
boxplot(cells2.trt$residuals) #plot boxplot of residuals
plot(cells2.trt) #display residuals versus fitter, normal QQ plot, leverage plot
```


```{r}
#CONTINUOUS EXPERIMENTAL PH DATA 
#Avtech_pH_data.csv
pH <- read.csv("RAnalysis/Data/Avtech_pH_data.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
pH$Date.Time <-as.POSIXct(pH$Date.Time, format="%m/%d/%y %H:%M") #convert date format
pH$Date <- as.Date(pH$Date.Time) #convert Date only
pH$Time <- format(as.POSIXct(pH$Date.Time) ,format = "%H:%M:%S") #convert Time only

pH.low <- do.call(data.frame,aggregate(Low ~ Date, data = pH, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
pH.med <- do.call(data.frame,aggregate(Medium ~ Date, data = pH, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
pH.amb <- do.call(data.frame,aggregate(Ambient ~ Date, data = pH, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
pH.low$Treatment <- "Super Low" #Add treatment Information
colnames(pH.low) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
pH.med$Treatment <- "Low" #Add treatment Information
colnames(pH.med) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
pH.amb$Treatment <- "Ambient" #Add treatment Information
colnames(pH.amb) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
daily.pH <- rbind(pH.amb, pH.med, pH.low) #bind treatment data 
daily.pH #view data

# Plot daily averages of pH data for the complete experiment
All.pH <- ggplot(daily.pH, aes(x=Date, y=mean, group=Treatment)) + #set up plot information
  geom_errorbar(aes(ymin=daily.pH$mean-daily.pH$se, ymax=daily.pH$mean+daily.pH$se), colour="black", width=.1, position = position_dodge(width = 0.05)) + #add standard error bars about the mean
  geom_point(aes(shape=Treatment), size = 2, position = position_dodge(width = 0.05)) + #include points in the shape of the treatments
  annotate("text", x=as.Date("2016-03-25"), y=7.95, label = "No Data") + #add text to the graphic where data are missing
  annotate("text", x=as.Date("2016-06-15"), y=7.95, label = "No Data") + #add text to the graphic where data are missing
  xlab("Time") + #label x axis
  ylab("pH Total Scale") + # label y axis
  ylim(6.8,8.1) + # set y axis scale
  geom_vline(xintercept = 16899, linetype="dotted", color = "gray", size=1) + #add vertical line
  geom_vline(xintercept = 16928, linetype="dashed", color = "gray", size=1) + #add vertical line
  geom_vline(xintercept = 17013, linetype="dotted", color = "gray", size=1) + #add vertical line
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(), #Set the plot background
        legend.position=c(.65, .25), #set legend location
        legend.text = element_text(size = 8), #set the legend text size
        legend.key = element_blank(), #remove the legend background
        legend.title = element_text(size=8, face="bold")) + #set legend title attributes
  ggtitle("A) Experimental pH") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) #set title attributes
All.pH #view plot
```

```{r}
#CONTINUOUS EXPERIMENTAL TEMPERATURE DATA
#Avtech_temp_data.csv
temp <- read.csv("RAnalysis/Data/Avtech_temp_data.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
temp$Date.Time <-as.POSIXct(temp$Date.Time, format="%m/%d/%y %H:%M") #convert date format
temp$Date <- as.Date(temp$Date.Time) #convert Date only
temp$Time <- format(as.POSIXct(temp$Date.Time) ,format = "%H:%M:%S") #convert time only

temp.low <- do.call(data.frame,aggregate(Low ~ Date, data = temp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
temp.med <- do.call(data.frame,aggregate(Medium ~ Date, data = temp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
temp.amb <- do.call(data.frame,aggregate(Ambient ~ Date, data = temp, function(x) c(mean = mean(x), se = std.error(x)))) #calculate mean and sem of each treatment by Day
temp.low$Treatment <- "Super Low" #Add treatment Information
colnames(temp.low) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
temp.med$Treatment <- "Low" #Add treatment Information
colnames(temp.med) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
temp.amb$Treatment <- "Ambient" #Add treatment Information
colnames(temp.amb) <- c("Date", "mean", "se", "Treatment") #rename columns to generic format
daily.temp <- rbind(temp.amb, temp.med, temp.low) #bind treatment data 
daily.temp #view data

All.temp <- ggplot(daily.temp, aes(x=Date, y=mean, group=Treatment)) + #set up plot information
  geom_errorbar(aes(ymin=daily.temp$mean-daily.temp$se, ymax=daily.temp$mean+daily.temp$se), colour="black", width=.1, position = position_dodge(width = 0.05)) + #add standard error bars about the mean
  geom_point(aes(shape=Treatment), size = 2, position = position_dodge(width = 0.05)) + #include points in the shape of the treatments
  annotate("text", x=as.Date("2016-05-25"), y=14, label = "No Data") + #add text to the graphic where data are missing
  xlab("Time") + #label x axis
  ylab("Temperature °C") + #label y axis
  ylim(0,20) + # set y axis scale
  geom_vline(xintercept = 16899, linetype="dotted", color = "gray", size=1) + #add vertical line
  geom_vline(xintercept = 16928, linetype="dashed", color = "gray", size=1) + #add vertical line
  geom_vline(xintercept = 17013, linetype="dotted", color = "gray", size=1) + #add vertical line
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(), #Set the plot background
        legend.position = "none") + #remove legend
  ggtitle("B) Experimental Temperature") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) #set title attributes
All.temp #view plot

FigureS1 <- arrangeGrob(All.pH, All.temp, ncol=1)
ggsave(file="RAnalysis/Figs/Fig.S1.pdf", FigureS1, width = 6, height = 8, units = c("in"))
```

```{r}
#SEAWATER CHEMISTRY ANALYSIS FOR DISCRETE MEASUREMENTS

#pH Tris Calibration Curves
#Data to calculate conversion equations from mV to total scale using tris standard for pH probe
path <-("RAnalysis/Data/pH_Calibration_Files/") #set path to calibration file folder
file.names<-list.files(path = path, pattern = "csv$") #list all the file names with csv 
pH.cals <- data.frame(matrix(NA, nrow=length(file.names), ncol=4, dimnames=list(file.names,c("Date", "Intercept", "Slope","R2")))) #generate an empty 3 column dataframe with specific column names

for(i in 1:length(file.names)) { # for every file in list start at the first and run this following function
  Calib.Data <-read.table(file.path(path,file.names[i]), header=TRUE, sep=",", na.string="NA", as.is=TRUE) #reads in the data files
  model <-lm(mVTris ~ TTris, data=Calib.Data) #runs a linear regression of mV as a function of temperature
  coe <- coef(model) #extracts the coeffecients
  R <- summary(model)$r.squared #extracts the R2
  pH.cals[i,2:3] <- coe #inserts coef in the dataframe
  pH.cals[i,4] <- R #inserts R2 in the dataframe
  pH.cals[i,1] <- substr(file.names[i],1,8) #stores the file name in the Date column
}

colnames(pH.cals) <- c("Calib.Date",  "Intercept",  "Slope", "R2") #names the columns of the dataframe
pH.cals #view data

#Prepare input file for seacarb
SW.chem <- read.csv("RAnalysis/Data/SW_Chem_Juvenile_Geoduck.csv", header=TRUE, sep=",", na.strings="NA") #load data with a header, separated by commas, with NA as NA
SW.chem <- merge(pH.cals, SW.chem, by="Calib.Date") #merge pH calibrations with Seawater chemistry data
R <- 8.31447215 #gas constant in J mol-1 K-1 
F <-96485.339924 #Faraday constant in coulombs mol-1
mvTris <- SW.chem$Temperature*SW.chem$Slope+SW.chem$Intercept #calculate the mV of the tris standard using the temperature mv relationships in the measured standard curves 
STris<-27.5 #salinity of the Tris from local batch made for Manchester WA salinity conditions in Spring 2016. Batch date = 20160214
phTris<- (11911.08-18.2499*STris-0.039336*STris^2)*(1/(SW.chem$Temperature+273.15))-366.27059+ 0.53993607*STris+0.00016329*STris^2+(64.52243-0.084041*STris)*log(SW.chem$Temperature+273.15)-0.11149858*(SW.chem$Temperature+273.15) #calculate the pH of the tris (Dickson A. G., Sabine C. L. and Christian J. R., SOP 6a)
SW.chem$pH.Total<-phTris+(mvTris/1000-SW.chem$pH.MV/1000)/(R*(SW.chem$Temperature+273.15)*log(10)/F) #calculate the pH on the total scale (Dickson A. G., Sabine C. L. and Christian J. R., SOP 6a)
SW.chem <- na.omit(SW.chem) #remove NAs


#Calculate CO2 parameters using seacarb
carb.output <- carb(flag=8, var1=SW.chem$pH.Total, var2=SW.chem$TA/1000000, S= SW.chem$Salinity, T=SW.chem$Temperature, P=0, Pt=0, Sit=0, pHscale="T", kf="pf", k1k2="l", ks="d") #calculate seawater chemistry parameters using seacarb
carb.output$ALK <- carb.output$ALK*1000000 #convert to µmol kg-1
carb.output$CO2 <- carb.output$CO2*1000000 #convert to µmol kg-1
carb.output$HCO3 <- carb.output$HCO3*1000000 #convert to µmol kg-1
carb.output$CO3 <- carb.output$CO3*1000000 #convert to µmol kg-1
carb.output$DIC <- carb.output$DIC*1000000 #convert to µmol kg-1
carb.output <- cbind(SW.chem$Measure.Date,  SW.chem$Exposure, SW.chem$Tank,  SW.chem$Treatment, carb.output) #combine the sample information with the seacarb output
colnames(carb.output) <- c("Date", "Exposure", "Tank",  "Treatment",	"flag",	"Salinity",	"Temperature","Patm",	"Pressure",	"pH",	"CO2",	"fCO2", "pCO2",	"fCO2pot","pCO2pot","fCOinsitu", "pCOinsitu","HCO3",	"CO3",	"DIC", "TA",	"Aragonite.Sat", 	"Calcite.Sat") #Rename columns to describe contents
carb.output <- subset(carb.output, select= c("Exposure","Date",  "Tank",  "Treatment",	"Salinity",	"Temperature",		"pH",	"CO2",	"pCO2",	"HCO3",	"CO3",	"DIC", "TA",	"Aragonite.Sat"))

#Calculate descriptive stats for seawater chemistry by Exposure
ccarb <- melt(carb.output[,c(1,3,4,5:14)], id.vars=c("Exposure", "Treatment", "Tank")) #reshape data into long format
Exp1 <-subset(ccarb, Exposure == "Exposure1") #separate out exposure 1 for all data
Exp2 <-subset(ccarb, Exposure == "Exposure2") #separate out exposure 2 for all data

SWC.Tanks <- ddply(ccarb, c("Exposure", "Tank", "variable"), summarise, #apply functions to sewater chem data
      N = length(na.omit(value)), #count the sample size removing NA
      mean = mean(value), #calculate average 
      sem = sd(value)/sqrt(N)) #calcualte the standard error of the mean

#Test for tank and treatment differences in Temperature and Total Alkalinity in Exposure 1
Exp1.Temp <-subset(Exp1, variable=="Temperature") #separate out exposure 1 for all data
temp1.tank <- aov(value ~Tank, data=Exp1.Temp) #test the hypothesis there is no difference in temperature between tanks
temp1.tank.res <-anova(temp1.tank) #view results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(temp1.tank$residuals) #plot histogram of residuals
boxplot(temp1.tank$residuals) #plot boxplot of residuals
plot(temp1.tank) #display residuals versus fitter, normal QQ plot, leverage plot

temp1.trt <- aov(value ~Treatment, data=Exp1.Temp) #test the hypothesis there is no difference in temperature between treatments
temp1.trt.res <- anova(temp1.trt) #statistical results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(temp1.trt$residuals) #plot histogram of residuals
boxplot(temp1.trt$residuals) #plot boxplot of residuals
plot(temp1.trt) #display residuals versus fitter, normal QQ plot, leverage plot

Exp1.TA <-subset(Exp1, variable=="TA") #separate out exposure 1 for all data
TA1.tank <- aov(value ~Tank, data=Exp1.TA) #test the hypothesis there is no difference in total alkalinity between tanks
TA1.tank.res <- anova(temp1.trt) #statistical results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(TA1.tank$residuals) #plot histogram of residuals
boxplot(TA1.tank$residuals) #plot boxplot of residuals
plot(TA1.tank) #display residuals versus fitter, normal QQ plot, leverage plot

TA1.trt <- aov(value ~Treatment, data=Exp1.TA) #test the hypothesis there is no difference in total alkalinity between treatments
TA1.trt.res <- anova(temp1.trt) #statistical results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(TA1.trt$residuals) #plot histogram of residuals
boxplot(TA1.trt$residuals) #plot boxplot of residuals
plot(TA1.trt) #display residuals versus fitter, normal QQ plot, leverage plot

#Test for tank and treatment differences in Temperature and Total Alkalinity in Exposure 2
Exp2.Temp <-subset(Exp2, variable=="Temperature") #separate out exposure 2 for all data
temp2.tank <- aov(value ~Tank, data=Exp2.Temp) #test the hypothesis there is no difference in temperature between tanks
temp2.tank.res <-anova(temp2.tank) #view results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(temp2.tank$residuals) #plot histogram of residuals
boxplot(temp2.tank$residuals) #plot boxplot of residuals
plot(temp2.tank) #display residuals versus fitter, normal QQ plot, leverage plot

temp2.trt <- aov(value ~Treatment, data=Exp2.Temp) #test the hypothesis there is no difference in temperature between treatments
temp2.trt.res <- anova(temp2.trt) #statistical results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(temp2.trt$residuals) #plot histogram of residuals
boxplot(temp2.trt$residuals) #plot boxplot of residuals
plot(temp2.trt) #display residuals versus fitter, normal QQ plot, leverage plot

Exp2.TA <-subset(Exp2, variable=="TA") #separate out exposure 2 for all data
TA2.tank <- aov(value ~Tank, data=Exp2.TA) #test the hypothesis there is no difference in total alkalinity between tanks
TA2.tank.res <- anova(temp2.trt) #statistical results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(TA2.tank$residuals) #plot histogram of residuals
boxplot(TA2.tank$residuals) #plot boxplot of residuals
plot(TA2.tank) #display residuals versus fitter, normal QQ plot, leverage plot

TA2.trt <- aov(value ~Treatment, data=Exp2.TA) #test the hypothesis there is no difference in total alkalinity between treatments
TA2.trt.res <- anova(temp2.trt) #statistical results
par(mfrow=c(3,2)) #set plotting configuration
par(mar=c(1,1,1,1)) #set margins for plots
hist(TA2.trt$residuals) #plot histogram of residuals
boxplot(TA2.trt$residuals) #plot boxplot of residuals
plot(TA2.trt) #display residuals versus fitter, normal QQ plot, leverage plot

#Calculate descriptive stats by Treatment
SWC.Treatments <- ddply(ccarb, c("Exposure", "Treatment", "variable"), summarise,
      N = length(na.omit(value)), #count the sample size removing NA
      mean = mean(value), #calculate average 
      sem = sd(value)/sqrt(N)) #calcualte the standard error of the mean

Exposure1 <-subset(SWC.Treatments, Exposure == "Exposure1") #separate out exposure 1
Exposure2 <-subset(SWC.Treatments, Exposure == "Exposure2") #separate out exposure 2
Exposure1.long <- reshape(Exposure1, idvar="Treatment", direction="wide", timevar = "variable", drop = c("Exposure", "N")) #reshape data format for table layout
Exposure2.long <- reshape(Exposure2, idvar="Treatment", direction="wide", timevar = "variable", drop = c("Exposure", "N")) #reshape data format for table layout
write.table (Exposure1.long, file="/Users/hputnam/MyProjects/Geoduck_Epi/project_juvenile_geoduck_oa/RAnalysis/Output/Seawater_chemistry_table_Output_Seed_exposure1.csv", sep=",", row.names = FALSE) #save data to output file
write.table (Exposure2.long, file="/Users/hputnam/MyProjects/Geoduck_Epi/project_juvenile_geoduck_oa/RAnalysis/Output/Seawater_chemistry_table_Output_Seed_exposure2.csv", sep=",", row.names = FALSE) #save data to output file

```


# Load Genomic Features and Sample Info

```{r}
#load sample information
sample.info <- read.csv("RAnalysis/Data/Sample.Info.csv", header=T, sep=",", na.string="NA", stringsAsFactors = F) #read in info
samp <- sample.info$Sample.ID # set sample info
samp <- gsub("[_]", "-", samp) #remove extra characters

#load genes gff
Genes <- read.csv("RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3", header=F, sep="\t", na.string="NA", skip=3, stringsAsFactors = F) #read in data file
Genes <- Genes[,c(1,4,5,9)] #select desired columns only
colnames(Genes) <- c("scaffold", "start", "stop", "gene") #rename columns
Genes$gene <- gsub(";.*","",Genes$gene) #remove extra characters
Genes$gene <- gsub("ID=","",Genes$gene) #remove extra characters

#Load annotation file
Annot <- read.csv("RAnalysis/Data/Genome/Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab", header=TRUE, sep="\t", na.string="NA", stringsAsFactors = F, skip=12)
colnames(Annot) <- c("gene","start", "end", "score", "uniprot.id", "Match.ID", "m_start", "m_end", "E.value", "identity", "al")
Annot$gene <- gsub("21910-","",Annot$gene) #remove extra characters
Annot$gene  <- gsub(".m01","",Annot$gene) #remove extra characters
Uniprots <- Annot[,c(1,5)] #pull out uniprot information
colnames(Uniprots) <- c("gene", "uniprot.id") #rename columns
write.csv(Uniprots, "RAnalysis/Data/Genome/Uniprot.IDS.csv") #save to file

#Obtain GO terms from Uniprot using Uniprot.IDS.csv https://www.uniprot.org/uploadlists/

GOs <- read.csv("RAnalysis/Data/Genome/Uniprot_GOs.csv", header=TRUE, sep=",", na.string="NA", stringsAsFactors = F) #read in gene ontology information
colnames(GOs) <- c("uniprot.id", "protein.name.short", "protein.name.long", "gene.names", "organism", "GO.IDs") #rename columns of gene ontology information
GO.data <- merge(Annot, GOs, by="uniprot.id")

```

# Filter and Concatenate Datasets

```{bash}
#Time dataset
mkdir RAnalysis/Data/OSF/Time 


multiIntersectBed -i \
RAnalysis/Data/OSF/Time/EPI-41_S38_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-42_S39_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-43_S40_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-44_S41_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-119_S31_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-120_S32_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-135_S35_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-136_S36_L005_5x.tab \
RAnalysis/Data/OSF/Time/EPI-151_S2_L002_5x.tab \
RAnalysis/Data/OSF/Time/EPI-152_S3_L002_5x.tab \
RAnalysis/Data/OSF/Time/EPI-153_S4_L002_5x.tab \
RAnalysis/Data/OSF/Time/EPI-154_S5_L002_5x.tab \
RAnalysis/Data/OSF/Time/EPI-181_S16_L003_5x.tab \
RAnalysis/Data/OSF/Time/EPI-182_S17_L003_5x.tab \
RAnalysis/Data/OSF/Time/EPI-184_S18_L003_5x.tab \
RAnalysis/Data/OSF/Time/EPI-185_S19_L003_5x.tab \
> RAnalysis/Data/OSF/Time/time.5x.bed

cat RAnalysis/Data/OSF/Time/time.5x.bed | awk '$4 ==16' > RAnalysis/Data/OSF/Time/time.filtered.16.5x.bed 

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file that ends in posOnly
#b: annotated gene list
#Save output in a new file that has the same base name and ends in -Annotated.txt
for f in RAnalysis/Data/OSF/Time/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in RAnalysis/Data/OSF/Time/*_gene
do
  intersectBed \
  -a ${f} \
  -b RAnalysis/Data/OSF/Time/time.filtered.16.5x.bed  \
  > ${f}_CpG_time.bed
done

wc -l RAnalysis/Data/OSF/Time/*_CpG_time.bed
```

```{bash}
# Day 10 Data Set
mkdir RAnalysis/Data/OSF/D10 

multiIntersectBed -i \
RAnalysis/Data/OSF/D10/EPI-103_S27_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-104_S28_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-111_S29_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-113_S30_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-119_S31_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-120_S32_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-127_S33_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-128_S34_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-135_S35_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-136_S36_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-143_S37_L005_5x.tab \
RAnalysis/Data/OSF/D10/EPI-145_S38_L005_5x.tab \
> RAnalysis/Data/OSF/D10/D10.5x.bed

cat RAnalysis/Data/OSF/D10/D10.5x.bed | awk '$4 ==12' > RAnalysis/Data/OSF/D10/D10.filtered.12.5x.bed 


#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in RAnalysis/Data/OSF/D10/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in RAnalysis/Data/OSF/D10/*_gene
do
  intersectBed \
  -a ${f} \
  -b RAnalysis/Data/OSF/D10/D10.filtered.12.5x.bed   \
  > ${f}_CpG_D10.bed
done

wc -l RAnalysis/Data/OSF/D10/*_CpG_D10.bed
```

```{bash}
# Day 135 Data

mkdir RAnalysis/Data/OSF/D135 

multiIntersectBed -i \
RAnalysis/Data/OSF/D135/EPI-151_S2_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-152_S3_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-153_S4_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-154_S5_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-159_S6_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-160_S7_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-161_S8_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-162_S9_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-167_S10_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-168_S11_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-169_S12_L002_5x.tab \
RAnalysis/Data/OSF/D135/EPI-170_S13_L002_5x.tab \
> RAnalysis/Data/OSF/D135/D135.5x.bed

cat RAnalysis/Data/OSF/D135/D135.5x.bed | awk '$4 ==12' > RAnalysis/Data/OSF/D135/D135.filtered.12.5x.bed 

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in RAnalysis/Data/OSF/D135/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in RAnalysis/Data/OSF/D135/*_gene
do
  intersectBed \
  -a ${f} \
  -b RAnalysis/Data/OSF/D135/D135.filtered.12.5x.bed   \
  > ${f}_CpG_D135.bed
done

wc -l RAnalysis/Data/OSF/D135/*_CpG_D135.bed
```

```{bash}
# Day 145 Data

mkdir RAnalysis/Data/OSF/D145 

multiIntersectBed -i \
RAnalysis/Data/OSF/D145/EPI-175_S14_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-176_S15_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-181_S16_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-182_S17_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-184_S18_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-185_S19_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-187_S20_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-188_S21_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-193_S22_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-194_S23_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-199_S24_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-200_S25_L003_5x.tab \
RAnalysis/Data/OSF/D145/EPI-205_S26_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-206_S27_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-208_S28_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-209_S29_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-214_S30_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-215_S31_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-220_S32_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-221_S33_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-226_S34_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-227_S35_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-229_S36_L004_5x.tab \
RAnalysis/Data/OSF/D145/EPI-230_S37_L004_5x.tab \
> RAnalysis/Data/OSF/D145/D145.5x.bed

cat RAnalysis/Data/OSF/D145/D145.5x.bed | awk '$4 ==24' > RAnalysis/Data/OSF/D145/D145.filtered.24.5x.bed 

#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to annotated genes
#wb: Print all lines in the second file
#a: file 
#b: annotated gene list
#Save output in a new file that has the same base name 
for f in RAnalysis/Data/OSF/D145/*5x.tab
do
  intersectBed \
  -wb \
  -a ${f} \
  -b RAnalysis/Data/Genome/Panopea-generosa-v1.0.a4.gene.gff3 \
  > ${f}_gene
done

#intersect with file to subset only those positions found in all samples
for f in RAnalysis/Data/OSF/D145/*_gene
do
  intersectBed \
  -a ${f} \
  -b RAnalysis/Data/OSF/D145/D145.filtered.24.5x.bed   \
  > ${f}_CpG_D145.bed
done

wc -l RAnalysis/Data/OSF/D145/*_CpG_D145.bed
```

# Load in methylation counts

```{r}
# Time
meth.data.T <- list.files(path = "RAnalysis/Data/OSF/Time", pattern = "time.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = FALSE) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.T) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.T$gene <- gsub(";.*","",meth.data.T$gene) #remove extra characters
meth.data.T$gene <- gsub("ID=","",meth.data.T$gene) #remove extra characters
meth.data.T$Sample.ID <- gsub("RAnalysis/Data/OSF/Time/","",meth.data.T$Sample.ID) #remove extra characters
meth.data.T$Sample.ID <- gsub("_.*","",meth.data.T$Sample.ID) #remove extra characters 
meth.data.T$Sample.ID <- gsub("-","_",meth.data.T$Sample.ID) #remove extra characters

MD.Time <- merge(meth.data.T,sample.info, by="Sample.ID")
```

```{r}
#Day 10
meth.data.D10 <- list.files(path = "RAnalysis/Data/OSF/D10", pattern = "D10.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = FALSE) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)

colnames(meth.data.D10) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D10$gene <- gsub(";.*","",meth.data.D10$gene) #remove extra characters
meth.data.D10$gene <- gsub("ID=","",meth.data.D10$gene) #remove extra characters
meth.data.D10$Sample.ID <- gsub("RAnalysis/Data/OSF/D10/","",meth.data.D10$Sample.ID) #remove extra characters
meth.data.D10$Sample.ID <- gsub("_.*","",meth.data.D10$Sample.ID) #remove extra characters 
meth.data.D10$Sample.ID <- gsub("-","_",meth.data.D10$Sample.ID) #remove extra characters

MD.D10 <- merge(meth.data.D10, sample.info, by="Sample.ID")
```

```{r}
#Day 135
meth.data.D135 <- list.files(path = "RAnalysis/Data/OSF/D135", pattern = "D135.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = FALSE) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.D135) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D135$gene <- gsub(";.*","",meth.data.D135$gene) #remove extra characters
meth.data.D135$gene <- gsub("ID=","",meth.data.D135$gene) #remove extra characters
meth.data.D135$Sample.ID <- gsub("RAnalysis/Data/OSF/D135/","",meth.data.D135$Sample.ID) #remove extra characters
meth.data.D135$Sample.ID <- gsub("_.*","",meth.data.D135$Sample.ID) #remove extra characters 
meth.data.D135$Sample.ID <- gsub("-","_",meth.data.D135$Sample.ID) #remove extra characters

MD.D135 <- merge(meth.data.D135, sample.info, by="Sample.ID")
```

```{r}
#Day 145
meth.data.D145 <- list.files(path = "RAnalysis/Data/OSF/D145", pattern = "D145.bed$", full.names=TRUE) %>%
  set_names(.) %>% 
  map_dfr(read.csv,.id="Sample.ID", header=FALSE, sep="\t", na.string="NA", stringsAsFactors = FALSE) %>% 
  dplyr::select(-c(V3,V7:V14)) %>%
  group_by(Sample.ID)
colnames(meth.data.D145) <- c("Sample.ID", "scaffold", "position","per.meth","meth","unmeth","gene")

meth.data.D145$gene <- gsub(";.*","", meth.data.D145$gene) #remove extra characters
meth.data.D145$gene <- gsub("ID=","",meth.data.D145$gene) #remove extra characters
meth.data.D145$Sample.ID <- gsub("RAnalysis/Data/OSF/D145/","",meth.data.D145$Sample.ID) #remove extra characters
meth.data.D145$Sample.ID <- gsub("_.*","",meth.data.D145$Sample.ID) #remove extra characters 
meth.data.D145$Sample.ID <- gsub("-","_",meth.data.D145$Sample.ID) #remove extra characters

MD.D145 <- merge(meth.data.D145, sample.info, by="Sample.ID")
```

```{r}

MD.ALL <- rbind(MD.Time, MD.D10, MD.D135, MD.D145)
MD.ALL.Genes <- unique(MD.ALL$gene)
write.csv(MD.ALL.Genes, file = "RAnalysis/Output/MD.ALL.Genes_For_GO_Background.csv")
```

# Testing for Differentially Methylated Genes

```{r}
# Comparison of Ambient samples through Time
# Binomial GLM to test for differentially methylated genes

sub_meth_table  <- MD.Time
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to stored results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ TimePoint, 
             data=sub_meth_table1, family=binomial)
  a <- anova(fit, test="Chisq")
  
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3], #uncomment if you want to include position of CpG within a gene
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4], #uncomment if you want to include position of CpG within a gene interaction with treatment
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}

results[is.na(results)] <- 0

results$adj.pval.TimePoint <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH') #uncomment if you want to include position of CpG within a gene
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH') #uncomment if you want to include position of CpG within a gene interaction with treatment

Time.Amb.sig <-results
Time.Amb.sig <- Time.Amb.sig[,c(1,3)]
Time.Amb.sig <- Time.Amb.sig[order(Time.Amb.sig$adj.pval.TimePoint),]
Time.Amb.sig <- Time.Amb.sig[which(Time.Amb.sig$adj.pval.TimePoint<0.05), ]

#write.table(Time.Amb.sig, '/home/hputnam/MyPorjects/Geoduck_Meth/RAnalysis/Output/Time_sig.tsv', sep='\t', row.names=FALSE)

sub_meth_table$per.meth <- (sub_meth_table$meth/(sub_meth_table$meth+sub_meth_table$unmeth))*100 #calculate percent methylation

# Annotation of DMG under ambient conditions through time
Time.Amb.annot <- merge(Time.Amb.sig , Annot, by="gene", all.x=TRUE)
Time.Amb.annot <- merge(Time.Amb.annot, GOs, by="uniprot.id", all.x=TRUE)
Time.Amb.annot <- Time.Amb.annot[order(Time.Amb.annot$adj.pval.TimePoint),]
Time.Amb.annot <- Time.Amb.annot[!duplicated(Time.Amb.annot$gene),]
write.table(Time.Amb.annot, 'RAnalysis/Output/Table_S1_Time_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Ambient samples through Time
##### GO enrichment of DMGs #####
ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(Time.Amb.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.Time <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.Time[is.na(GO.Time)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.Time$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.Time$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.TIME <- subset(enriched.GO.05, ontology=="MF")
MF.TIME <- MF.TIME[order(-MF.TIME$numDEInCat),]
CC.TIME <- subset(enriched.GO.05, ontology=="CC")
CC.TIME <- CC.TIME[order(-CC.TIME$numDEInCat),]
BP.TIME <- subset(enriched.GO.05, ontology=="BP")
BP.TIME <- BP.TIME[order(-BP.TIME$numDEInCat),]
write.csv(MF.TIME , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_TIME.csv")
write.csv(CC.TIME , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_TIME.csv")
write.csv(BP.TIME , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_TIME.csv")
```


```{r}
# Comparison of Treatments at Day 10
# Binomial GLM to test for differentially methylated genes

meth_table  <- MD.D10
sub_meth_table <- meth_table[meth_table$Initial.Treatment == 'Ambient' | meth_table$Initial.Treatment == 'Low' | meth_table$Initial.Treatment == 'Super.Low', ]
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)
  
#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to stored results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment, 
             data=sub_meth_table1, family=binomial, maxit = 100)
  a <- anova(fit, test="Chisq")
  
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3], #uncomment if you want to include position of CpG within a gene
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4], #uncomment if you want to include position of CpG within a gene interaction with treatment
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}

results[is.na(results)] <- 0

results$adj.pval.Initial.Treatment <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH') #uncomment if you want to include position of CpG within a gene
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH') #uncomment if you want to include position of CpG within a gene interaction with treatment

D10.sig <-results
D10.sig <- D10.sig[,c(1,3)]
D10.sig <- D10.sig[order(D10.sig$adj.pval.Initial.Treatment),]
D10.sig <- D10.sig[which(D10.sig$adj.pval.Initial.Treatment<0.05), ]

write.table(D10.sig, 'RAnalysis/Output/D10_sig.tsv', sep='\t', row.names=FALSE)

# Annotation of D10 DMG 
D10.sig.annot <- merge(D10.sig, Annot, by="gene", all.x=TRUE)
D10.sig.annot <- merge(D10.sig.annot, GOs, by="uniprot.id", all.x=TRUE)
D10.sig.annot <- D10.sig.annot[order(D10.sig.annot$adj.pval.Initial.Treatment),]
D10.sig.annot <- D10.sig.annot[!duplicated(D10.sig.annot$gene),]
write.table(D10.sig.annot, 'RAnalysis/Output/Table_S2_D10_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 10
##### GO enrichment of DMGs #####
ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(D10.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D10 <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.D10[is.na(GO.D10)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D10$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D10$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector <- as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.D10 <- subset(enriched.GO.05, ontology=="MF")
MF.D10 <- MF.D10[order(-MF.D10$numDEInCat),]
CC.D10 <- subset(enriched.GO.05, ontology=="CC")
CC.D10 <- CC.D10[order(-CC.D10$numDEInCat),]
BP.D10 <- subset(enriched.GO.05, ontology=="BP")
BP.D10 <- BP.D10[order(-BP.D10$numDEInCat),]
write.csv(MF.D10 , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_D10.csv")
write.csv(CC.D10 , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_D10.csv")
write.csv(BP.D10 , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_D10.csv")

D10.nuc.genes <- D10.sig.annot[grep("GO:0004518", D10.sig.annot$GO.IDs), ]
D10.galac.genes <- D10.sig.annot[grep("GO:0001733", D10.sig.annot$GO.IDs), ]

```

```{r}
# Comparison of Treatments at Day 135
# Binomial GLM to test for differentially methylated genes

meth_table  <- MD.D135
sub_meth_table <- meth_table[meth_table$Initial.Treatment == 'Ambient' | meth_table$Initial.Treatment == 'Low' | meth_table$Initial.Treatment == 'Super.Low', ]

sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to store results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
  
  #subset the dataframe gene by gene
  sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
  
  #if(length(unique(sub_meth_table1$position)) >1){
  
  # fit glm position model
  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment, 
             data=sub_meth_table1, family=binomial)
  #} else {
  #  # fit glm model
  #  fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment * position, 
  #             data=sub_meth_table1, family=binomial)
  #}
  #s <- step(fit, trace=0)
  a <- anova(fit, test="Chisq")
  a
  # capture summary stats to data frame
  df <- data.frame(gene = sub_meth_table1[1,7],
                   pval.treatment = a$`Pr(>Chi)`[2],
                   #pval.position = a$`Pr(>Chi)`[3],
                   #pval.treatment_x_position = a$`Pr(>Chi)`[4],
                   stringsAsFactors = F)
  
  # bind rows of temporary data frame to the results data frame
  results <- rbind(results, df)
  
}

results[is.na(results)] <- 0

results$adj.pval.treatment <- p.adjust(results$pval.treatment, method='BH')
#results$adj.pval.position <- p.adjust(results$pval.position, method='BH')
#results$adj.pval.treatment_x_position <- p.adjust(results$pval.treatment_x_position, method='BH')

D135.sig <-results
D135.sig <- D135.sig[,c(1,3)]
D135.sig <- D135.sig[order(D135.sig$adj.pval.treatment),]
D135.sig <- D135.sig[which(D135.sig$adj.pval.treatment<0.05), ]
write.table(D135.sig, 'RAnalysis/Output/D135_sig.tsv', sep='\t', row.names=FALSE)

# Annotation of D135 DMG 
D135.sig.annot <- merge(D135.sig, Annot, by="gene", all.x=TRUE)
D135.sig.annot <- merge(D135.sig.annot, GOs, by="uniprot.id", all.x=TRUE)
D135.sig.annot <- D135.sig.annot[order(D135.sig.annot$adj.pval.treatment),]
D135.sig.annot <- D135.sig.annot[!duplicated(D135.sig.annot$gene),]
write.table(D135.sig.annot, 'RAnalysis/Output/Table_S3_D135_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 135
##### GO enrichment of DMGs #####
ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(D135.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D135 <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.D135[is.na(GO.D135)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D135$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D135$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.D135 <- subset(enriched.GO.05, ontology=="MF")
MF.D135 <- MF.D135[order(-MF.D135$numDEInCat),]
CC.D135 <- subset(enriched.GO.05, ontology=="CC")
CC.D135 <- CC.D135[order(-CC.D135$numDEInCat),]
BP.D135 <- subset(enriched.GO.05, ontology=="BP")
BP.D135 <- BP.D135[order(-BP.D135$numDEInCat),]
write.csv(MF.D135 , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_D135.csv")
write.csv(CC.D135 , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_D135.csv")
write.csv(BP.D135 , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_D135.csv")

D135.nuc.genes <- D135.sig.annot[grep("GO:0004518", D135.sig.annot$GO.IDs), ]
D135.galac.genes <- D135.sig.annot[grep("GO:0001733", D135.sig.annot$GO.IDs), ]

```

```{r}
# Comparison of Treatments at Day 145
# Binomial GLM to test for differentially methylated genes
meth_table <- MD.D145
sub_meth_table <- meth_table[meth_table$Initial.Treatment == "Ambient" | meth_table$Initial.Treatment == "Low" | meth_table$Initial.Treatment == "Super.Low"
                             | meth_table$Secondary.Treatment == "Ambient" | meth_table$Secondary.Treatment == "Low", ]
sub_meth_table$group <- paste0(sub_meth_table$Sample.ID, sub_meth_table$gene)

#filter for genes with >5 methylated positions
min.filt <- count(sub_meth_table, vars = c( group))
newdata <- min.filt[ which(min.filt$n > 4), ]
sub_meth_table <- sub_meth_table[sub_meth_table$group %in% newdata$vars,]

# create data frame to store results
results <- data.frame()
gs <- unique(sub_meth_table$gene)

#first subset the unique dataframes and second run the GLMs
for(i in 1:length(sub_meth_table$gene)){
    
    #subset the dataframe gene by gene
    sub_meth_table1 <- subset(sub_meth_table, gene ==gs[i])
    
    # fit glm position model
    fit <- glm(matrix(c(meth, unmeth), ncol=2) ~ Initial.Treatment * Secondary.Treatment, 
               data=sub_meth_table1, family=binomial)

    a <- anova(fit, test="LRT")
    
    # capture summary stats to data frame
    df <- data.frame(gene = sub_meth_table1[1,7],
                     pval.treatment1 = a$`Pr(>Chi)`[2],
                     pval.treatment2 = a$`Pr(>Chi)`[3],
                     #position = a$`Pr(>Chi)`[2], 
                     pval.treatment1_x_treatment2 = a$`Pr(>Chi)`[4],
                     #pval.treatment1_x_position = a$`Pr(>Chi)`[2],
                     #pval.treatment2_x_position = a$`Pr(>Chi)`[2],
                     #pval.treatment1_x_pval.treatment2_x_position = a$`Pr(>Chi)`[2],
                     stringsAsFactors = F)

    # bind rows of temporary data frame to the results data frame
    results <- rbind(results, df)
    
  }

results[is.na(results)] <- 0

results$adj.pval.treatment1 <- p.adjust(results$pval.treatment1, method='BH')
results$adj.pval.treatment2 <- p.adjust(results$pval.treatment2, method='BH')
results$adj.pval.treatment1_x_treatment2 <- p.adjust(results$pval.treatment1_x_treatment2, method='BH')
#results$adj.pval.treatment1_x_position <- p.adjust(results$pval.treatment1_x_position, method='BH')
#results$adj.pval.treatment2_x_position <- p.adjust(results$pval.treatment2_x_position, method='BH')
#results$adj.pval.treatment1_x_pval.treatment2_x_position <- p.adjust(results$pval.treatment1_x_pval.treatment2_x_position, method='BH')

D145.sig <-results
D145.sig <- D145.sig[,c(1,5:7)]
D145.sig <- D145.sig[order(results$adj.pval.treatment1_x_treatment2),]
D145.sig <- D145.sig[which(D145.sig$adj.pval.treatment1_x_treatment2<0.05), ]
write.table(D145.sig, 'RAnalysis/Output/D145_sig.tsv', sep='\t', row.names=FALSE)

sub_meth_table$per.meth <- (sub_meth_table$meth/(sub_meth_table$meth+sub_meth_table$unmeth))*100
sub_meth_table.D145 <- sub_meth_table

# Annotation of Secondary Exposure Interaction DMG 
D145.sig.annot <- merge(D145.sig, Annot, by="gene", all.x=TRUE)
D145.sig.annot <- merge(D145.sig.annot, GOs, by="uniprot.id", all.x=TRUE)
D145.sig.annot <- D145.sig.annot[order(D145.sig.annot$adj.pval.treatment1_x_treatment2),]
D145.sig.annot <- D145.sig.annot[!duplicated(D145.sig.annot$gene),]
write.table(D145.sig.annot, 'RAnalysis/Output/Table_S4_D145_sig_annot.tsv', sep='\t', row.names=FALSE)
```

```{r}
# GO Enrichment Analysis of Treatment at Day 145
##### GO enrichment of DMGs #####
ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(D145.sig$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.D145 <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.D145[is.na(GO.D145)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.D145$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.D145$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.D145 <- subset(enriched.GO.05, ontology=="MF")
MF.D145 <- MF.D145[order(-MF.D145$numDEInCat),]
CC.D145 <- subset(enriched.GO.05, ontology=="CC")
CC.D145 <- CC.D145[order(-CC.D145$numDEInCat),]
BP.D145 <- subset(enriched.GO.05, ontology=="BP")
BP.D145 <- BP.D145[order(-BP.D145$numDEInCat),]
write.csv(MF.D145 , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_D145.csv")
write.csv(CC.D145 , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_D145.csv")
write.csv(BP.D145 , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_D145.csv")

D145.nuc.genes <- D145.sig.annot[grep("GO:0004518", D145.sig.annot$GO.IDs), ]
D145.galac.genes <- D145.sig.annot[grep("GO:0001733", D145.sig.annot$GO.IDs), ]

```


```{r}
##### DMGs in all comparisons #####

ints_10_135 <- merge(D10.sig, D135.sig, by="gene")
ints_10_145 <- merge(D10.sig, D145.sig, by="gene")
ints_135_145 <- merge(D135.sig, D145.sig, by="gene")
ints_all <- merge(ints_10_135, D145.sig, by="gene")
ints_all.annot <- merge(ints_all, Annot, by="gene", all.x=TRUE)
ints_all.annot <- ints_all.annot[!duplicated(ints_all.annot$gene),]
write.table(ints_all.annot, 'RAnalysis/Output/Table_S5_DMGs_consistent_INT.tsv', sep='\t', row.names=FALSE)

```

```{r}
# GO Enrichment Analysis of DMG present in all comparisons
##### GO enrichment of DMGs #####

#ALL= should be all possible genes in D0, D10, D135, and D145

ALL<-as.data.frame(MD.ALL.Genes) #set the all gene list
colnames(ALL) <-c("gene")
DMG <- as.character(ints_all$gene) #set the enrichment test list
LENGTH <-cbind(Genes$gene,Genes$stop-Genes$start) #reads in a list of gene lengths
colnames(LENGTH) <-c("gene", "length") #name columns
LENGTH <- merge(LENGTH, ALL, by="gene") #merge gene names and length
GO.INT <- merge(ALL, GO.data,by="gene", all=T) #merge GO, ID
GO.INT[is.na(GO.INT)] <- "unknown" #list NA as unknown
splitted <- strsplit(as.character(GO.INT$GO.IDs), ";") #slit into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.INT$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DMG.vector <-c(t(DMG))
ID.vector <- LENGTH$gene
LENGTH.vector <-LENGTH$length

gene.vector=as.integer(ALL.vector%in%DMG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ALL.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, 
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")

MF.INT <- subset(enriched.GO.05, ontology=="MF")
MF.INT <- MF.INT[order(-MF.INT$numDEInCat),]
CC.INT <- subset(enriched.GO.05, ontology=="CC")
CC.INT <- CC.INT[order(-CC.INT$numDEInCat),]
BP.INT <- subset(enriched.GO.05, ontology=="BP")
BP.INT <- BP.INT[order(-BP.INT$numDEInCat),]
write.csv(MF.INT , file = "RAnalysis/Output/MF_Sig_Enriched_GO.05_INT.csv")
write.csv(CC.INT , file = "RAnalysis/Output/CC_Sig_Enriched_GO.05_INT.csv")
write.csv(BP.INT , file = "RAnalysis/Output/BP_Sig_Enriched_GO.05_INT.csv")
```

```{r}
##### Heatmap of DMG in Ambient by time #####
Time.Amb.meth_table.means <- aggregate(per.meth ~ TimePoint*gene, data=MD.Time, FUN=mean)
Time.Amb.meth_table.means <- Time.Amb.meth_table.means[Time.Amb.meth_table.means$gene %in% Time.Amb.sig$gene,]

Time.Amb <- Time.Amb.meth_table.means %>% pivot_wider(names_from = TimePoint, values_from = per.meth)
#MD.Time.1 <- MD.Time[,c(1,4,7,12)]

#Time.Amb <- reshape(MD.Time.1, 
#  timevar = "Sample.ID",
#  idvar = c("gene"),
#  direction = "wide")

#Time.Amb.mat <- Time.Amb[,c(26,28,30,32,2,4,6,8,10,12,14,16,18,20,22,24)]
Time.Amb.mat <- Time.Amb[,-1]-rowMeans(Time.Amb[,-1])

pheatmap(Time.Amb.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =TRUE, cluster_cols=FALSE,
         show_colnames =TRUE) #plot heatmap of all DEG by group
dev.off()

pdf("RAnalysis/Output/Time.DMG_Abs.pdf")
pheatmap(Time.Amb.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =FALSE, cluster_cols=FALSE,
         show_colnames =TRUE) #plot heatmap of all DEG by group
dev.off()
```

```{r}
D10.meth_table.means <- aggregate(per.meth ~ Initial.Treatment*gene, data=MD.D10, FUN=mean)
D10.meth_table.means <- D10.meth_table.means[D10.meth_table.means$gene %in% ints_all$gene,]

#D10 <- D10.meth_table.means %>% pivot_wider(names_from = Initial.Treatment, values_from = per.meth)
#D10 <- D10[D10$gene %in% ints_all.annot$gene,]

D10 <- reshape(D10.meth_table.means, 
  timevar = "Initial.Treatment",
  idvar = c("gene"),
  direction = "wide")

row.names(D10) <- D10$gene

D10.mat <- D10[,2:4]
D10.mat <- D10.mat[order(D10.mat$per.meth.Ambient),]
D10.mat <- D10.mat-rowMeans(D10.mat)

Breaks <- seq(min(0), max(100), length = 100)

pdf("RAnalysis/Output/D10.DMG_Abs_Meth.pdf", width = 4, height= 4)
pheatmap(D10.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =TRUE, cluster_cols=FALSE,cluster_rows = TRUE,fontsize_row = 2,
         show_colnames =TRUE) #plot heatmap of all DEG by group
dev.off()
```

```{r}
D135.meth_table.means <- aggregate(per.meth ~ Initial.Treatment*gene, data=MD.D135, FUN=mean)
D135.meth_table.means <- D135.meth_table.means[D135.meth_table.means$gene %in% ints_all$gene,]

D135 <- reshape(D135.meth_table.means, 
  timevar = "Initial.Treatment",
  idvar = c("gene"),
  direction = "wide")

row.names(D135) <- D135$gene

D135.mat <- D135[,2:4]
D135.mat <- D135.mat[match(rownames(D10.mat), rownames(D135.mat)),]
#D135.mat <- D135.mat-rowMeans(D135.mat[1])

pdf("RAnalysis/Output/D135.DMG_Abs_Meth.pdf", width = 1.5, height= 4)
pheatmap(D135.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =TRUE, cluster_cols=FALSE,cluster_rows = FALSE,fontsize_row = 2,
         show_colnames =TRUE, breaks = Breaks) #plot heatmap of all DEG by group
dev.off()
```

```{r}
D145.meth_table.means <- aggregate(per.meth ~ gene*Initial.Treatment*Secondary.Treatment, data=MD.D145, FUN=mean)
D145.meth_table.means <- D145.meth_table.means[D145.meth_table.means$gene %in% ints_all$gene,]
D145.meth_table.means$group <- paste0(D145.meth_table.means$Initial.Treatment, D145.meth_table.means$Secondary.Treatment)
D145.meth_table.means <- D145.meth_table.means[,c(1,4,5)]
#D145 <- D145.meth_table.means %>% pivot_wider(names_from = c(treatment1,treatment2), values_from = per.meth)
D145 <- reshape(D145.meth_table.means, 
  timevar = "group",  
  idvar = c("gene"),
  direction = "wide")

row.names(D145) <- D145$gene
D145.mat <- D145[,2:7]

# D145.mat$per.meth.AmbientAmbient <- D145.mat$per.meth.AmbientAmbient  - D145.mat$per.meth.AmbientAmbient
# D145.mat$per.meth.AmbientLow <- D145.mat$per.meth.AmbientLow  - D145.mat$per.meth.AmbientAmbient
# D145.mat$per.meth.LowAmbient <- D145.mat$per.meth.LowAmbient  - D145.mat$per.meth.LowAmbient
# D145.mat$per.meth.LowLow <- D145.mat$per.meth.LowLow  - D145.mat$per.meth.LowAmbient
# D145.mat$per.meth.Super.LowAmbient <- D145.mat$per.meth.Super.LowAmbient  - D145.mat$per.meth.Super.LowAmbient
# D145.mat$per.meth.Super.LowLow <- D145.mat$per.meth.Super.LowLow  - D145.mat$per.meth.Super.LowAmbient
col.order <- c("per.meth.AmbientAmbient", "per.meth.AmbientLow", "per.meth.LowAmbient", "per.meth.LowLow", "per.meth.Super.LowAmbient", "per.meth.Super.LowLow")
D145.mat  <- D145.mat[,col.order]
D145.mat <- D145.mat[match(rownames(D10.mat), rownames(D145.mat)),]

D145.mat.1 <- D145.mat[,c(1:2)]
D145.mat.2 <- D145.mat[,c(3:4)]
D145.mat.3 <- D145.mat[,c(5:6)]

pdf("RAnalysis/Output/D145.DMG_Abs_Meth.pdf", width = 3, height= 5)
pheatmap(D145.mat, clustering_method = "average", 
         clustering_distance_rows="euclidean", show_rownames =TRUE, cluster_rows=FALSE, cluster_cols=FALSE, gaps_col = c(2,4),
         fontsize_row = 2, show_colnames =TRUE, breaks = Breaks) #plot heatmap of all DEG by group
dev.off()


# pdf("/home/hputnam/MyPorjects/Geoduck_Meth/RAnalysis/Output/D145.DMG_Rel_Amb_Meth_1.pdf")
# pheatmap(D145.mat.1, clustering_method = "average", 
#          clustering_distance_rows="euclidean", show_rownames =TRUE, fontsize_row = 2, cluster_cols=FALSE,
#          cluster_rows=F, show_colnames =TRUE) #plot heatmap of all DEG by group
# dev.off()
# pdf("/home/hputnam/MyPorjects/Geoduck_Meth/RAnalysis/Output/D145.DMG_Rel_Amb_Meth_2.pdf")
# pheatmap(D145.mat.2, clustering_method = "average", 
#          clustering_distance_rows="euclidean", show_rownames =TRUE, fontsize_row = 2, cluster_cols=FALSE,
#          cluster_rows=F, show_colnames =TRUE) #plot heatmap of all DEG by group
# dev.off()
# pdf("/home/hputnam/MyPorjects/Geoduck_Meth/RAnalysis/Output/D145.DMG_Rel_Amb_Meth_3.pdf")
# pheatmap(D145.mat.3, clustering_method = "average", 
#          clustering_distance_rows="euclidean", show_rownames =TRUE, fontsize_row = 2, cluster_cols=FALSE,
#          cluster_rows=F, show_colnames =TRUE) #plot heatmap of all DEG by group
# dev.off()
```


```{r}

ints_all.annot <- na.omit(ints_all.annot)

for (i in 1:nrow(ints_all.annot)){
  sig.gene <- subset(MD.D10, gene ==ints_all.annot$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/D10plot_", ints_all.annot$gene[i],"_",ints_all.annot$protein.name[i],".png"), width = 14, height = 10, units = "cm")
}

for (i in 1:nrow(ints_all.annot)){
  sig.gene <- subset(MD.D135, gene ==ints_all.annot$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/D135plot_", ints_all.annot$gene[i],"_",ints_all.annot$protein.name[i],".png"), width = 14, height = 10, units = "cm")
}

for (i in 1:nrow(ints_all.annot)){
  sig.gene <- subset(sub_meth_table.D145, gene ==ints_all.annot$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(data=means, aes(x=Secondary.Treatment, y=per.meth, group=Initial.Treatment, colour=Initial.Treatment, shape=Initial.Treatment)) + #plot data
    geom_line(size=0.7, position=position_dodge(.1)) + #plot lines
    scale_colour_manual(values = c("blue", "purple", "red")) + #set line color
    geom_point(size=4, position=position_dodge(.1), colour="black") + #set point characteristics
    scale_shape_manual(values=c(1,18,16)) + #set shapes
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), #plot error bars
                  width=0, position=position_dodge(.1), colour="black") + #set error bar characteristics
    xlab("Secondary Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/D145plot_",ints_all.annot$gene[i],"_", ints_all.annot$protein.name[i],".png"), width = 14, height = 10, units = "cm")
}
```

```{r}

#Patterns of GO enrichment genes at each time point

#Day 10 
for (i in 1:nrow(D10.nuc.genes)){
  sig.gene <- subset(MD.D10, gene==D10.nuc.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    ylim(0,100)+ #set y limits to 0 and 100%
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/NUC_D10plot_", D10.nuc.genes$gene[i],"_",D10.nuc.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}

for (i in 1:nrow(D10.galac.genes)){
  sig.gene <- subset(MD.D10, gene==D10.galac.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    ylim(0,100)+ #set y limits to 0 and 100%
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/Galac_D10plot_", D10.galac.genes$gene[i],"_",D10.galac.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}

#Day 135 
for (i in 1:nrow(D135.nuc.genes)){
  sig.gene <- subset(MD.D135, gene==D135.nuc.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    ylim(0,100)+ #set y limits to 0 and 100%
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/NUC_D135plot_", D135.nuc.genes$gene[i],"_",D135.nuc.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}

for (i in 1:nrow(D135.galac.genes)){
  sig.gene <- subset(MD.D135, gene==D135.galac.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*gene, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(means, aes(x=Initial.Treatment, y=per.meth, fill=Initial.Treatment)) +
    geom_bar(stat="identity", color="black",
             position=position_dodge()) +
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), width=.2,
                  position=position_dodge(.9)) +
    xlab("Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    ylim(0,100)+ #set y limits to 0 and 100%
    scale_fill_manual(values = c("blue", "purple", "red"))+
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/Galac_D135plot_", D135.galac.genes$gene[i],"_",D135.galac.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}

#Day 145 
for (i in 1:nrow(D145.nuc.genes)){
  sig.gene <- subset(MD.D145, gene ==D145.nuc.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(data=means, aes(x=Secondary.Treatment, y=per.meth, group=Initial.Treatment, colour=Initial.Treatment, shape=Initial.Treatment)) + #plot data
    geom_line(size=0.7, position=position_dodge(.1)) + #plot lines
    scale_colour_manual(values = c("blue", "purple", "red")) + #set line color
    geom_point(size=4, position=position_dodge(.1), colour="black") + #set point characteristics
    scale_shape_manual(values=c(1,18,16)) + #set shapes
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), #plot error bars
                  width=0, position=position_dodge(.1), colour="black") + #set error bar characteristics
    xlab("Secondary Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/NUC_D145plot_", D145.nuc.genes$gene[i],"_",D145.nuc.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}

for (i in 1:nrow(D145.galac.genes)){
  sig.gene <- subset(MD.D145, gene ==D145.galac.genes$gene[i])
  means <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=mean)
  ses <- aggregate(per.meth ~ Initial.Treatment*Secondary.Treatment, data=sig.gene, FUN=std.error)
  means$ses <- ses$per.meth

  temp_plot <- ggplot(data=means, aes(x=Secondary.Treatment, y=per.meth, group=Initial.Treatment, colour=Initial.Treatment, shape=Initial.Treatment)) + #plot data
    geom_line(size=0.7, position=position_dodge(.1)) + #plot lines
    scale_colour_manual(values = c("blue", "purple", "red")) + #set line color
    geom_point(size=4, position=position_dodge(.1), colour="black") + #set point characteristics
    scale_shape_manual(values=c(1,18,16)) + #set shapes
    geom_errorbar(aes(ymin=per.meth-ses, ymax=per.meth+ses), #plot error bars
                  width=0, position=position_dodge(.1), colour="black") + #set error bar characteristics
    xlab("Secondary Treatment") + #plot x axis label
    ylab("Percent Methylation") + #plot y axis label
    theme_bw()

  ggsave(temp_plot, file=paste0("RAnalysis/Output/Galac_D145plot_", D145.galac.genes$gene[i],"_",D145.galac.genes$protein.name.short[i],".png"), width = 14, height = 10, units = "cm")
}

```














